---
author-meta:
- Jordan A. Berg
- Youjia Zhou
- T. Cameron Waller
- Yeyun Ouyang
- Sara M. Nowinski
- Tyler Van Ry
- Ian George
- James E. Cox
- "Bei Wang<sup>\u2020</sup>"
- "Jared Rutter<sup>\u2020</sup>"
bibliography:
- content/manual-references.json
date-meta: '2020-09-12'
header-includes: "<!--\nManubot generated metadata rendered from header-includes-template.html.\nSuggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html\n-->\n<meta name=\"dc.format\" content=\"text/html\" />\n<meta name=\"dc.title\" content=\"Gazing into the Metaboverse: Automated exploration and contextualization of metabolic data\" />\n<meta name=\"citation_title\" content=\"Gazing into the Metaboverse: Automated exploration and contextualization of metabolic data\" />\n<meta property=\"og:title\" content=\"Gazing into the Metaboverse: Automated exploration and contextualization of metabolic data\" />\n<meta property=\"twitter:title\" content=\"Gazing into the Metaboverse: Automated exploration and contextualization of metabolic data\" />\n<meta name=\"dc.date\" content=\"2020-09-12\" />\n<meta name=\"citation_publication_date\" content=\"2020-09-12\" />\n<meta name=\"dc.language\" content=\"en-US\" />\n<meta name=\"citation_language\" content=\"en-US\" />\n<meta name=\"dc.relation.ispartof\" content=\"Manubot\" />\n<meta name=\"dc.publisher\" content=\"Manubot\" />\n<meta name=\"citation_journal_title\" content=\"Manubot\" />\n<meta name=\"citation_technical_report_institution\" content=\"Manubot\" />\n<meta name=\"citation_author\" content=\"Jordan A. Berg\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-5096-0558\" />\n<meta name=\"twitter:creator\" content=\"@jordanberg0\" />\n<meta name=\"citation_author\" content=\"Youjia Zhou\" />\n<meta name=\"citation_author_institution\" content=\"School of Computing, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Scientific Computing and Imaging Institute, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-4501-8496\" />\n<meta name=\"citation_author\" content=\"T. Cameron Waller\" />\n<meta name=\"citation_author_institution\" content=\"Division of Medical Genetics, Department of Medicine, School of Medicine, University of California San Diego\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-0313-6646\" />\n<meta name=\"citation_author\" content=\"Yeyun Ouyang\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0001-9523-1044\" />\n<meta name=\"citation_author\" content=\"Sara M. Nowinski\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-4744-6101\" />\n<meta name=\"citation_author\" content=\"Tyler Van Ry\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Metabolomics Core Facility, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0000-0000-0000\" />\n<meta name=\"citation_author\" content=\"Ian George\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0003-1361-694X\" />\n<meta name=\"citation_author\" content=\"James E. Cox\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Diabetes &amp; Metabolism Research Center, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Metabolomics Core Facility, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-5977-2350\" />\n<meta name=\"citation_author\" content=\"Bei Wang&lt;sup&gt;\u2020&lt;/sup&gt;\" />\n<meta name=\"citation_author_institution\" content=\"School of Computing, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Scientific Computing and Imaging Institute, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-9240-0700\" />\n<meta name=\"citation_author\" content=\"Jared Rutter&lt;sup&gt;\u2020&lt;/sup&gt;\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biochemistry, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Diabetes &amp; Metabolism Research Center, University of Utah\" />\n<meta name=\"citation_author_institution\" content=\"Howard Hughes Medical Institute, University of Utah\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-2710-9765\" />\n<meta property=\"og:type\" content=\"article\" />\n<meta property=\"twitter:card\" content=\"summary_large_image\" />\n<link rel=\"icon\" type=\"image/png\" sizes=\"192x192\" href=\"https://manubot.org/favicon-192x192.png\" />\n<link rel=\"mask-icon\" href=\"https://manubot.org/safari-pinned-tab.svg\" color=\"#ad1457\" />\n<meta name=\"theme-color\" content=\"#ad1457\" />\n<!-- end Manubot generated metadata -->"
keywords:
- metabolomics
- metabolism
- network
- omics
- rna-seq
- proteomics
- gene
- protein
- metabolite
- context
lang: en-US
manubot-clear-requests-cache: false
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
title: 'Gazing into the Metaboverse: Automated exploration and contextualization of metabolic data'
...

## Authors


[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-5096-0558)
Jordan A. Berg<sup>1</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-4501-8496)
Youjia Zhou<sup>2,3</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-0313-6646)
T. Cameron Waller<sup>4</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0001-9523-1044)
Yeyun Ouyang<sup>1</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-4744-6101)
Sara M. Nowinski<sup>1</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0000-0000-0000)
Tyler Van Ry<sup>1,5</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0003-1361-694X)
Ian George<sup>1</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-5977-2350)
James E. Cox<sup>1,5,6</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-9240-0700)
Bei Wang<sup>†</sup><sup>2,3</sup>,
[![ORCID icon](images/orcid.svg){height="11px" width="11px"}](https://orcid.org/0000-0002-2710-9765)
Jared Rutter<sup>†</sup><sup>1,6,7</sup>
<small>

1. Department of Biochemistry, University of Utah
2. School of Computing, University of Utah
3. Scientific Computing and Imaging Institute, University of Utah
4. Division of Medical Genetics, Department of Medicine, School of Medicine, University of California San Diego
5. Metabolomics Core Facility, University of Utah
6. Diabetes & Metabolism Research Center, University of Utah
7. Howard Hughes Medical Institute, University of Utah
</small>


To whom correspondence should be addressed: jordan.berg@biochem.utah.edu; rutter@biochem.utah.edu

<sup>†</sup> Senior Author


## Abstract {.page_break_before}

Metabolism and its component reactions are complex, each with variable inputs, outputs, and modifiers. The harmony between these factors consequently determines the health and stability of a cell or an organism. Perturbations to any reaction unit or set of reactions can have rippling downstream effects, which can be challenging to trace across the total reaction network, particularly when the effects occur between canonical representations of pathways. Researchers have primarily utilized reductionist approaches to understand metabolic reaction systems; however, customary methods often limit the analysis scope. Even systems-centric omics approaches can be limited when only a handful of high magnitude signals in the data are prioritized. To address these challenges, we developed Metaboverse, an interactive tool for the exploration and automated extraction of potential regulatory events, patterns, and trends from multi-omic data within the context of the metabolic network and other reaction networks. This framework will be foundational in increasing our ability to holistically understand static, temporal, and multi-condition metabolic events and perturbations as well as gene-metabolite intra-cooperativity. Metaboverse is freely available under a GPL-3.0 license at [https://github.com/Metaboverse/](https://github.com/Metaboverse/).


## Introduction

Metabolism is a complex network of reactions and interactions between genes, enzymes, protein complexes, and metabolites [@doi:10.3390/metabo9040076]. To understand these complex components, researchers frequently adopt reductionist approaches to tease apart the characteristics and mechanics of these processes and determine how they fit into the larger picture of biology and disease. Such approaches are a vital component in the scientific process, but a reductionist approach may miss many essential properties of metabolism. For example, in differential gene expression analysis, researchers rely on thresholds of magnitude and statistical significance to prioritize genes for follow-up study. However, this can inadvertently limit the scope of study of metabolism when, in fact, metabolism is a highly interconnected system where distal components and their modulation can have coordinated or rippling effects across the network [@doi:10.1016/j.coisb.2018.10.001; @doi:10.1038/s41540-018-0067-y]. Biological perturbations involve complex, cooperative effects, many of which may seem negligible in isolation. The current approach is analogous to telling the story of Little Red Riding Hood by reading only the 20 most frequent words. Indeed, doing so efficiently highlights keywords like "wolf" and "little red riding hood," but also prevents a coherent story from being told. This approach makes it difficult for someone who had never heard the story of Little Red Riding Hood to comprehend the plot (Figure @fig:little_red).

![
  **Word cloud of 20 most frequent words or phrases in the story of Little Red Riding Hood.**
  Text taken from the original telling of Little Red Riding Hood by Charles Perrault. Words and phrases derived using [@url:https://github.com/amueller/word_cloud]. Outline derived from the piece, "Little Red Riding Hood" by J.W. Smith.
](./content/figures/red_riding.png "Tall image"){#fig:little_red height=4in}

Several computational tools have risen to prominence over the past decade to resolve these issues in data contextualization. We will highlight four such representative and popular tools for their respective properties, though many more exist [@doi:10.1530/JME-18-0055]. First is MetaboAnalyst, which relies heavily on set enrichment methods for the analysis of data, or examining the belongingness of sets of significantly changed analytes (i.e., metabolite, protein, or gene measurements), for extracting interesting information. Network visualization is available, but it focuses primarily on interaction networks, and its ability to extract regulatory information is limited, particularly in an automated fashion [@doi:10.1093/nar/gkp356; @doi:10.1093/nar/gky310]. Second is Cytoscape, which serves as a general go-to platform for representing biological or other networks. One strength of Cytoscape is the ability to design apps or plug-ins to develop customized analyses; however, comprehensive and metabolism-specific regulatory identification methods are unavailable [@doi:10.1101/gr.1239303]. One plug-in for Cytoscape that focuses on metabolic data is MetScape, but again, this tool is generally limited to pathway enrichment, correlation networks, and data visualization and does not integrate approaches to identify regulatory mechanisms within the data [@doi:10.1093/bioinformatics/btx012; @doi:10.1093/bioinformatics/btr661; @doi:10.1093/bioinformatics/btq048]. MetExplore focuses on the curation of networks and is particularly useful for collaborative annotation of emerging models of organisms with incomplete metabolic network curations. It additionally can layer experimental data on the network for visualization [@doi:10.1093/nar/gky301; @doi:10.1093/nar/gkq312]. A companion tool to MetExplore is MetExploreViz, which enables interactive and flexible visualization of omics data on metabolic networks [@doi:10.1093/bioinformatics/btx588]. Reactome, which our tool uses for the curation of biological networks, also offers analytical tools for user data, but again relies on set enrichment or manual methods for identifying patterns. While all have their respective utility, there is still a need for tools that integrate these features and automate pattern and trend detection across metabolic networks to extract regulatory and other features from data [@doi:10.1093/nar/gki072; @pmid:31691815; @pmid:29145629]. This need is particularly pronounced in common cases where experimental data have sparse coverage of the metabolic network, a common situation in modern metabolomics.

To address these limitations in current conventions of metabolic data analysis, contextualization, and interpretation, we created Metaboverse, a cross-platform application to aid users in understanding and interpreting their model's metabolic story and generating new hypotheses from the available information. Metaboverse is an interactive tool for exploratory data analysis that searches user data in the context of the metabolic network to identify unusual patterns and trends within the data. By identifying these trends and patterns across the global reaction network, Metaboverse will aid scientists in formulating new hypotheses from their data and in designing follow-up experiments for a deeper understanding of their model (Figure @fig:overview).

![
  **Metaboverse conceptual overview.**
  Illustration summarizing the usage of Metaboverse to model biological data on the global reaction network to rapidly identify regulatory hotspots. Traditionally, when scientists perform an omics experiment, they tend to focus on a few features that are differentially regulated. Metaboverse allows a user to input multiple omics data types, which it layers upon the metabolic network. Metaboverse then uses this integrated model to identify patterns of putative regulatory potential in the data. The user can also dynamically explore metabolic pathways and other network representations.
](./content/figures/graphical_abstract.png "Square image"){#fig:overview}

To provide a robust platform for the exploration of single or multi-omic metabolic data, we developed several computational features to address the aims discussed above. We developed a pattern search engine for the rapid and automated identification of different patterns and trends in omics data on the metabolic network. Conceptually, this search engine borrows principles of topological motif searching from graph theory [@doi:10.1126/science.1089167; @doi:10.1038/ng881] and builds upon the concept of "activity motifs" [@doi:10.1038/nbt.1499]. Briefly, this method analyzes each reaction in the network and identifies all those that match a given pattern, such as the average values of the substrates versus the average values of the products being greater than 2-fold different.

Another feature introduced in Metaboverse allows for the interactive exploration of specific reactions or reaction entities with on-the-fly pattern search analysis. The user can explore specific pathways of interest and look for other interesting patterns and trends within these pathways. Perturbations in the abundance or behavior of a particular metabolic component can lead to downstream gene expression changes and other phenotypic modulations in a biological system. Users can therefore select a given component and explore the upstream and downstream reaction neighbors across the total reaction network. Users can also select any component from a drop-down menu and immediately view its reaction neighborhood.

We also include an interactive perturbation exploration module, which displays all the perturbed reactions from the selected super-pathway. For example, if the user were to display all perturbed reactions in Metabolism, naturally, all connected reactions that are sequentially perturbed would connect, while more disconnected components would appear isolated and the user could then explore the extent of perturbations to their system.

Finally, we tackle the challenge of sparsity, particularly in metabolomics datasets. We designed a reaction collapsing feature that supports reaction pattern searches across multiple reactions, where intermediate steps of the reaction trajectory may not have measured values for their respective input or output components. This feature is particularly important in allowing for pattern detection across the missing components of a pathway. For example, the human metabolic network alone consists of over 1000 metabolite nodes, but an average gas chromatography-mass spectrometry (GC-MS) experiment may only quantify 100-200 metabolites by name. By bridging proximal reactions, one can detect patterns that would require the comparison of the terminal ends of that sub-pathway. This reaction collapsing feature can help the user identify more inconspicuous patterns within the global reaction network.

Currently, Metaboverse accepts standard two-condition experiments and timecourse or multi-condition experiments. The user can provide any combination of datasets from each omics category (values mapping to gene, protein, and metabolite identifiers). Timecourse inputs can consist of matched samples for every time point or a combination of steady-state levels and time points. The user provides fold change and statistical measures with appropriate multiple-testing adjustments from their respective omics datasets, which Metaboverse then layers onto the metabolic network. Metaboverse accepts data from various model organisms, including humans, mice, yeast, and zebrafish. The foundational curation of Metaboverse is built on the Reactome reaction curations, providing 90+ species that the Metaboverse environment can process. To validate the features and methods available in Metaboverse, we analyzed a variety of datasets to provide representative vignettes that highlight the reliability of Metaboverse for extracting canonical features, and its power for identifying novel features and patterns and generate new hypotheses. We expect that Metaboverse will become a foundational tool in the analytical toolkit and augment the user's ability to more holistically explore and contextualize metabolic reaction networks.


## Results

### Metaboverse is a dynamic, user-friendly tool for the exploration of high-throughput biological data in organism-specific pathways

#### Overview
We designed Metaboverse as a lightweight, stand-alone app for the dynamic exploration of high-throughput biological data. The pathway curations are derived from Reactome [@doi:10.1093/nar/gkz1031], coupled with additional metabolite synonym cross-referencing and mapping from the ChEBI [@doi:10.1093/nar/gkv1031] and Human Metabolome (HMDB) [@doi:10.1093/nar/gkx1089] databases. As of the time of writing, Metaboverse is capable of analyzing data for over 90 species. A user begins by providing the desired output location for a new curation and organism of interest. If the user has previously curated data onto a reaction network for the organism of interest, they can alternatively provide this intermediate file. Next, the user provides the relevant transcriptomic, proteomic, and/or metabolomic datasets to layer onto the global reaction network of their organism of interest. Input data types can be extended to any dataset that uses the relevant mapping IDs; for example, one could provide ribosome profiling translation efficiency values mapped to the appropriate gene IDs for layering onto the network and downstream analysis. For each omics type, the user provides fold change and statistical values for each measurement. For example, if a user-provided data for a two-condition comparison transcriptomics dataset, these values would consist of log<sub>2</sub>(fold change) values between the experimental and wild-type conditions, and the appropriate statistical values. As transcriptomics count data follows a negative binomial distribution, the appropriately adjusted p-values based on multiple hypothesis testing for this distribution should be used [@doi:10.1186/s13059-014-0550-8, @doi:10.1093/bioinformatics/btp616]. For other omics types, these multiple hypothesis testing methods will vary based on the data distribution and other conventions. During the data input step, the user will also specify a few experimental parameters for consideration during downstream analysis and visualization. Following these user inputs, the organism's network is curated and the data is overlaid upon the network. A database file for the dataset is output for future analysis (see Figure @fig:package_overview).

![
  **Metaboverse package overview.**
  An illustration outlining the data input, processing, and exploration steps of the Metaboverse package. The user provides the name of the organism of interest from a drop-down menu along with an output location. The user then has the option to provide transcriptomics, proteomics, and or metabolomics datasets. These datasets can be single- or multi-condition or timecourse experiments. Data is formatted as follows: row names are the measured entity names or IDs, the first column is log<sub>2</sub>(fold change) or other measurement values, and the second column is statistical measurements. During this step, the user can also provide sample labels and other modifiers to customize the curation of the data on the reaction network. Metaboverse will then build the model. Once the model is complete, the user will be able to visualize the patterns identified within reactions, explore pathway-specific or global perturbation networks, and perform general pathway and nearest reaction neighborhood exploration of the data.
](./content/figures/package_overview.png "Square image"){#fig:package_overview}

We chose to format network visualizations in a reaction-centric format, which means that a sub-network unit is comprised of a reaction node, linked with its reaction input and reaction output nodes, along with any modifier nodes. If modifier nodes are a protein complex, the component proteins of that complex are linked to the complex node. Any genes available in the database are linked to their related proteins. Each node and link-type are color-coded to help the user quickly distinguish the details of a reaction sub-network structure. Reactions with matching input or output sides are naturally linked, allowing a multi-reaction pathway to emerge from this network structure. This formatting differs from other formatting styles the user may be familiar with, such as KEGG pathway maps, for various reasons. As this dynamic visualization approach differs from more classical methods, each pathway within the visualization module contains a button linking to a more classically formatted representation of the pathway for reference sourced from Reactome [@doi:10.1093/nar/gkz1031]. While classical pathway definitions are used to group sets of reactions during visualization, analytical methods such as the pattern search in Metaboverse explore the global reaction network and display the pathways as flexible networks. This approach aids in making pattern analysis within the reaction network less subject to biases arising from constraining analysis to only familiar pathways and allows for exploration of patterns that may not neatly fit into canonical pathway representations.

#### Handling data sparsity within the global biological reaction network
Missing data points, particularly in metabolomics experiments, can be frequent and make the analysis of pathways and identification of regulatory patterns in the network challenging. Thousands of metabolites are known to participate in human metabolism, but the current state of the technology used for determining which mass spectra belong to which metabolite can be time consuming and often results in a limited quantification of data points. This limitation can lead to gaps in the measured metabolic network, leading to challenges when exploring and analyzing the data. For example, the human metabolic network is comprised of over 1000 metabolites. In contrast, the average GC-MS experiment may measure only 100-200 metabolites by name. This can confound pattern recognition across reactions. Therefore, we developed a reaction compression algorithm (described and visualized in the Methods section) that collapses up to three reactions with missing data points if they can be bridged with known data on the distal ends of the reaction path. Similar concepts have been used in metabolomic analysis before to identify amino acid-related metabolites [@doi:10.1038/s41586-019-1437-3]. However, we introduce the first computational implementation to our knowledge and adopt a slightly more conservative collapsing scheme. These bridged reactions, or pseudo-reactions, are visually distinct to show the user where reactions were collapsed and understand what that pseudo-reaction summarizes. We found that these collapsed reaction representations aided in identifying regulatory mechanisms across multiple reactions, which would have been missed by motif searches across traditional, uncollapsed representations of human metabolism (see Supplementary Figure @fig:supp_fig_collapse and accompanying text for a brief discussion of the benefits of this feature).

#### Rapid identification of interesting regulatory patterns in the reaction network
Following network curation, the user can visualize the available patterns identified across all reactions in the network. In a computational science context, a motif is a recurring pattern in network structure or the organization of network entities and their relationships. However, with omics data, we are more interested in identifying patterns involving expression or abundance of genes, proteins, and metabolites. Previous work by Checkik, et al. introduced the concept of "activity motifs." Instead of identifying motifs based on network structure, they identified motifs using the expression characteristics of nodes in transcription factor binding signaling networks [@doi:10.1038/nbt.1499]. We adapted this methodology to identify and interactively display interesting regulatory hotspots across all reactions in the network. For example, a reaction's inputs may exhibit increased levels and its outputs decreased levels, indicating a regulatory event occurring at the reaction.

Magnitude change and statistical values associated with each measurement for a measured entity are provided by the user. Once the data is layered upon the network, Metaboverse will consider all the reactions in the network and compare the expression or abundance patterns to those from a pre-defined library of regulatory patterns (please refer to the documentation for more details on available patterns [@url:https://metaboverse.readthedocs.io/en/latest]). Metaboverse then returns a graphical stamp view of conserved patterns and sorts these reactions based on the significance values associated with those components that determined the regulatory feature. Reactions can also be ordered based on the magnitude of change. We use a three-tiered sorting strategy when sorting by the associated statistical values. The highest prioritized reaction patterns are those where the relevant components that determined the presence of a regulatory pattern (inputs vs. outputs) are statistically significant. Of these reactions from the first tier, reactions are sorted by lowest to highest cumulative p-value or other relevant statistical value. In the second sorting tier, reactions with at least one side of the reaction with a statistically significant reaction component are sorted by statistical strength. Finally, all other reactions are sorted by the cumulative statistical value for the reaction components. The reaction's stamp shading indicates the sorting tier of each reaction, where green symbolizes that the components on both sides of the reaction passed significance thresholds, yellow symbolizes components on one side of the reaction passed significance thresholds, and gray symbolizes neither side of the reaction contained significant components.

During dynamic pathway visualization, reaction pattern nodes are visually distinct to quickly draw the user's attention to these interesting patterns. For a given regulatory pattern, the user can then explore each pathway where the reaction was found. Pattern analysis of the all reactions within the network will allow the user to rapidly identify interesting features within the data, particularly patterns between canonical pathway representations or within pathways that may seem tangential to their research. In the data vignettes below, we demonstrate this utility further.

#### Dynamic visualization of organism-specific reaction pathways
Following the curation of the global network as described above, the user can manually search individual pathways or individual entities and their reaction neighborhoods. For a given selection, all relevant reactions that are annotated as a part of that pathway will be visualized, along with their core input (substrate) and output (product) components. In addition to these core elements, known catalysts and inhibitors are included, as well as the component proteins, genes, and metabolites that form functional complexes involved in a particular reaction. For nodes representing protein complexes, simulated values are calculated by taking the average of all measured component entities in that complex. If gene values are available, but protein values are not, the user has the option to "broadcast" the gene values to the protein nodes. Assuming all statistical values are between 0 and 1, simulated statistics are calculated by taking the maximum statistic value of all measured component entities in that complex. In cases where a gene value is known, but its protein value is unmeasured, the protein value is optionally inferred using aggregated gene component values. Relevant pathway and analytical metadata are also displayed. All reaction pattern types are displayed using default thresholds if present in a selected pathway. Additional aids for visualization are also available. These aids include the ability to remove nodes from the visualization that contain a high number of relationships to other network features so that these nodes, which act as hubs in the network, do not lead to cluttered representations. These hub nodes are often ubiquitous features such as water and protons, which may be of limited interest to the user during data visualization [@doi:10.1093/gigascience/giz137]. Compartment domains are also visualized to show the cellular locations of reactions.

#### Visualization of downstream effects of network perturbations using reaction neighborhoods
The user may be interested in a particular metabolite or protein and the downstream effects their perturbation has on other reactions across all pathways. By double-clicking a node of interest or selecting the entity name from the drop-down menu, the user can explore all upstream and downstream effects across all pathways. The user can also define how many neighborhoods to display such that two or more reaction steps downstream or upstream of the selected entity are visualized [@doi:10.1007/PL00009293; @doi:10/bz69nr]. This functionality moves the analysis past traditional, pathway-centric approaches and contextualizes the far-reaching effects that the disruption of metabolism can have across classical pathways. However, as metabolism is a network with high connectivity [@doi:10.1093/gigascience/giz137], expanding past one reaction neighborhood without defining a hub threshold can effect performance. In these instances, Metaboverse will display a warning and suggest thresholds to use for the number of neighbors and hubs.

#### Exploring perturbation networks within the data
At times, it may be useful to a user to explore all perturbed reactions in a network. Metaboverse will display the total perturbation network for a given super-pathway selected by the user, where then the user is able to look across all pathways within fatty acid metabolism, for example, and identify any interesting features. During perturbation network display, all reactions are included and plotted if at least one component of that reaction passes the indicated magnitude or statistical threshold. Naturally, sequentially perturbed reactions will be stitched together in the resulting perturbation network.

### Data vignettes
To demonstrate the utility of Metaboverse, we analyzed both public and new metabolically-relevant datasets. From the vignettes provided below, we show that Metaboverse identifies points of interest previously described or expected and rapidly identifies unexpected and systematic regulatory patterns in a reaction network context. For an example of the reaction collapse algorithm's utility, we refer the reader to the associated supplement (Supplementary Figure @fig:supp_fig_collapse). Data from Vignette 1 is available in distributed versions of Metaboverse (as `test_data.zip`) to act as an example for users to familiarize themselves with the input data format.

#### Vignette 1: Perturbation of mitochondrial fatty acid synthesis across time
Mitochondrial fatty acid synthesis (mtFAS), an evolutionarily conserved pathway, has long been recognized to produce lipoic acid, a critical cofactor for several metabolic enzymes. Recent work has begun to uncover additional, important roles for this pathway. For example, we now know that mtFAS coordinates fatty acid synthesis with iron-sulfur (Fe-S) cluster biogenesis and assembly of oxidative phosphorylation complexes [@doi:10.1016/j.cub.2018.08.022; @doi:10.1016/j.molcel.2018.06.039; @doi:10.7554/eLife.17828]. Additionally, this pathway has received increased physiological focus with the discovery of patients with mutations in key mtFAS enzymes [@doi:10.1016/j.ajhg.2016.09.021].

The <i>Saccharomyces cerevisiae</i> Mct1 protein is an acyltransferase responsible for the transfer of a malonyl group from malonyl-CoA to the mitochondrial acyl carrier protein (ACP). By knocking out this gene, the activity of the mtFAS pathway is abolished. In order to probe the relationship between mtFAS-related protein concentration and the effects of its perturbation on downstream metabolic processes from a systematic perspective, we used an <i>mct1</i>&Delta; mutant <i>S. cerevisiae</i> strain. We previously performed steady-state comparative proteomics in this <i>mct1</i>&Delta; model 12 hours after shifting to a non-fermentable carbon source [@doi:10.1016/j.molcel.2018.06.039]. To complement these previous studies, we performed RNA-sequencing of transcript levels at 0, 3, and 12 hours after a shift to a non-fermentable carbon source and steady-state metabolomics at 0, 15, 30, 60, and 180 minutes after the shift in carbon source. By layering these data onto the <i>S. cerevisiae</i> metabolic network using Metaboverse, we observed interesting respiratory signatures as expected based on previous work [@doi:10.1111/mmi.12402; @doi:10.1016/j.molcel.2018.06.039]. For example, we noticed a strong pattern in the electron transfer from ubiquinol to cytochrome C via complex III of the electron transport chain (ETC) (Figure @fig:mct1_vignette A). At the protein level, the cytochrome C isoforms, CYC1 and CYC7, are both significantly reduced in concentration compared to wild-type cells (log<sub>2</sub>(fold change): -1.57 & -0.88; adjusted p-value: 1.21E-03 & 5.84E-03; Cohen's <i>d</i>: -7.99 & -3.94; respectively). This reduction in cytochrome C concentration is paired with a marked reduction in the concentration of the protein components of Complex III. These components catalyze the transfer of electrons from ubiquinol to cytochrome C (average log<sub>2</sub>(fold change) of all measured protein components: -2.03, where 9/11 component proteins were measured; 8/11 passed statistical significance thresholds of adjusted p-value <= 0.05. Range of significant log<sub>2</sub>(fold change) values: -0.34 to -3.21).

The second expected pattern of interest that was the marked reduction of TCA cycle-related enzymes. Reaction pattern analysis within the TCA cycle identified several putative regulatory hotspots between metabolites and metabolite-protein interactions throughout the TCA cycle (Figure @fig:mct1_vignette B). However, visualizing these identified reactions and data across the timecourse revealed interesting patterns. For this discussion, we will provide metabolite measurements at the 15-minute time point because the statistical strength for the discussed metabolites was generally strongest. For example, citrate levels decrease across the metabolomics timecourse (at 15 min; log<sub>2</sub>(fold change): -1.75, adjusted p-value: 6.61E-03; Cohen's <i>d</i>: -1.86), which is coincident with reduced steady-state levels of Ctp1 (log<sub>2</sub>(fold change): -0.64, adjusted p-value: 6.06E-03; Cohen's <i>d</i>: -4.73), a protein that catalyzes the transfer of citrate from the mitochondrial matrix to the cytosol [@doi:10.1074/jbc.270.8.4108]. Citrate is a key metabolite and the first step in the TCA cycle. We hypothesize that due to central carbon metabolite reductions, Ctp1 may be down-regulated in response to the decrease in citrate to maintain citrate concentrations within the mitochondria where it is perhaps most physiologically important in an acute respiratory metabolism context.

Another point of interest is the up-regulation of Dic1 (log<sub>2</sub>(fold change): 2.15, adjusted p-value: 5.95E-06; Cohen's <i>d</i>: 28.97), which catalyzes the exchange of dicarboxylates like malate (at 15 min; log<sub>2</sub>(fold change): 2.61, adjusted p-value: 3.23E-06; Cohen's <i>d</i>: 5.12) between the mitochondrial matrix and cytosol. Interestingly, Dic1 is essential for growth on non-fermentable carbon source media [@doi:10.1046/j.1365-2958.1999.01197.x]. When yeast, especially those with deficits in TCA cycle flux due to loss of Mct1, are switched to a non-fermentable carbon, as was done in our study, they might adapt by up-regulating Dic1, which we witness in this data.

![
  **Metaboverse identifies several reaction patterns of interest in <i>mct1</i>&Delta; cells.**
  (A) Steady-state proteomics (12 hours) overlaid on the reaction, "Electron transfer from ubiquinol to cytochrome c of complex III." (B) Steady-state proteomics (12 hours) and metabolomics at 15 minutes overlaid on TCA-related reactions. Appropriate time stamps for the data type are displayed in the upper-right hand corner of each subplot. Metabolomics and proteomics values are shown as node shading, where an increasingly blue shade indicates down-regulation, and an increasingly red shade indicates up-regulation. Measured log<2>(fold change) and statistical values for each entity are displayed below the node name. A gray node indicates a reaction. A bold gray node with a purple border indicates a potential regulatory pattern at this reaction for the given data type time points. Non-reaction circles indicate metabolites, squares indicate proteins or protein complexes, and ellipses indicate gene components. Gray edges are core relationships between reaction inputs and outputs. Green edges indicate a catalyst. Dashed blue edges point from a metabolite component to the complex in which it is involved. Dashed orange edges point from a protein component to the complex in which it is involved. Dashed purple edges point from a gene component to its protein product. Protein complexes with dashed borders indicate that the values displayed on that node were inferred from the constituent protein, metabolite, and gene measurements. The shading in sub-plot A demonstrates Metaboverse's ability to show compartmentalization, whereas sub-plot B has compartmentalization shading turned off.
](./content/figures/fig_mct1.png "Square image"){#fig:mct1_vignette}

We also noticed that although the TCA cycle enzymes are all reduced in concentration, several related metabolites are up-regulated across multiple time points in the dataset. It is possible, for example, that the increased fumarate levels (at 15 min; log<sub>2</sub>(fold change): 1.00, adjusted p-value: < 0.01; Cohen's <i>d</i>: 3.89) are related to the reduction in fumarate hydratase (FH tetramer, FUM1; log<sub>2</sub>(fold change): -1.39, adjusted p-value: < 0.01; Cohen's <i>d</i>: -4.91), as has been demonstrated previously and shown to associate with hereditary leiomyomatosis and renal cell cancer in humans [@doi:10.1038/nature19353]. These patterns would be an interesting point of further study to understand the adaptations these cells make to disruptions to the TCA cycle and the downstream consequences of these disruptions and adaptations.

By analyzing this multi-omics dataset using reaction pattern analysis and other interactive visualization available within Metaboverse, interesting questions arise. We see several reaction patterns that are expected based on prior knowledge of this biological model and other puzzling behaviors worthy of further follow-up. These identified patterns demonstrate the potential Metaboverse has to act as a valuable hypothesis-generation tool, particularly with multi-omics and timecourse datasets, and how this platform allows for better visualization and analysis of a user's dataset in the context of the metabolic or other reaction networks.

#### Vignette 2: Metabolic signatures in malignant human lung adenocarcinomas compared to nonmalignant lung tissue
We next turned to published human lung adenocarcinoma steady-state metabolomics data [@doi:10.1158/1940-6207.CAPR-14-0329] to assess the utility of Metaboverse when analyzing data. Lung cancer remains a leading cause of death worldwide, and an improved understanding of these tumors' metabolism is essential in understanding how to treat the disease better.

Consistent with the original study [@doi:10.1158/1940-6207.CAPR-14-0329] and our recent re-analysis of the data [@doi:10.1093/gigascience/giz137], nucleotide metabolism was broadly up-regulated in adenocarcinomas based on the reaction pattern and perturbation network analyses. In particular, the highest-scoring reaction pattern identified by Metaboverse involved the transformation of dc-adenosyl methionine and spermidine to form spermine and 5'-methylthioadenosine. This reaction is a step in polyamine metabolism, with ties to nucleotide metabolism (Figure @fig:lung_vignette A). We also notice similar perturbations in xanthine (log<sub>2</sub>(fold change): 1.35; adjusted p-value: 1.52E-05; Cohen's <i>d</i>: 0.86) (Figure @fig:lung_vignette B) and other metabolites, which are highlighted to the user quickly in the reaction pattern and perturbation network analyses. However, as previously emphasized [@doi:10.1093/gigascience/giz137], by approaching these perturbations in a reaction-centric approach, we can identify regulatory behavior that further contextualizes the data. For example, using a more manual approach, we previously highlighted the up-regulation of glyceric acid (log<sub>2</sub>(fold change): 0.50; adjusted p-value: 1.91E-04; Cohen's <i>d</i>: 0.73) coupled with the proximal down-regulation of 3-Phosphoglyceric acid (log<sub>2</sub>(fold change): -1.09; adjusted p-value: 4.74E-04; Cohen's <i>d</i>: -0.68) which could indicate the activity of glycerate kinase [@doi:10.1093/gigascience/giz137]. This connection was missed in the original study but was highlighted by Metaboverse's reaction pattern analysis module. This sub-pathway activity is consequential as it has connections to serine metabolism that contributes to generating an ideal environment for tumorigenesis (Figure @fig:lung_vignette C). Interestingly, of the measured and significant metabolites involved in the TCA cycle, we see a moderate up-regulation of malate (log<sub>2</sub>(fold change): 0.39, adjusted p-value: 3.85E-03; Cohen's <i>d</i>: 0.56) and down-regulation of citrate (log<sub>2</sub>(fold change): -0.64, adjusted p-value: 1.20E-03; Cohen's <i>d</i>: -0.63). This pattern is consistent with the hypothesis that the TCA cycle is starved of pyruvate-derived carbons and is compensating by shunting carbon into the TCA cycle via glutaminolysis. This compensatory pathway, a feature in many cancers, consists of glutamine transport into mitochondria and its conversion to alpha-ketoglutarate. Alpha-ketoglutarate then enters the second half of the TCA cycle to generate reducing equivalents, ATP, and biosynthetic intermediates [@pmid:429309; @doi:10.1146/annurev-bioeng-071516-044546]. Pyruvate is likely shunted towards the production of lactate and other biosynthetic products (no measurements for lactate were available in the dataset). This pattern might explain the lower citrate levels, while the second half of the TCA cycle could be fed through glutamate (log<sub>2</sub>(fold change): 0.47, adjusted p-value: 1.99E-06; Cohen's <i>d</i>: 0.98), providing the increased malate concentration (Figure @fig:lung_vignette D) [@doi:10.1038/nrc.2016.71].

Metaboverse simplifies the analysis process of metabolic data and allows the user more flexibility via an interactive platform that augments the identification of both canonical and novel regulatory patterns within the data. We were able to identify the regulatory pattern identified in Figure @fig:lung_vignette C previously using more manual approaches, but Metaboverse offers an automated platform for the discovery of such events not identified in the original study. We anticipate that the use of Metaboverse will enable the user to extract new and exciting hypotheses that can drive their fields forward.

![
  **Metaboverse identifies nucleotide metabolism and other signatures in lung adenocarcinoma metabolomics data.**
  (A) Up-regulation of nucleotide metabolites. (B) Identification of xanthine regulation by both motif and perturbation analysis. (C) Regulatory activity between glyceric acid and 3-Phosphoglyceric acid identified by Metaboverse's motif analysis. (D) Disruptions of TCA metabolism support canonical disruptions during adenocarcinoma development. Metabolomics values are shown as node shading, where an increasingly blue shade indicates down-regulation and an increasingly red shade indicates up-regulation. Measured log<sub>2</sub>(fold change) and statistical values for each entity are displayed below the node name. A gray node indicates a reaction. A bold gray node with a purple border indicates a motif at this reaction. Non-reaction circles indicate metabolites and squares indicate proteins or protein complexes. Gray edges indicate core relationships between reaction inputs and outputs. Green edges indicate a catalyst, and red edges indicate inhibitors. Dashed blue edges point from a metabolite component to the protein complex in which it is involved. Dashed orange edges point from a protein component to the protein complex in which it is involved. Protein complexes with dashed borders indicate that the values displayed on that node were inferred from the constituent protein and metabolite measurements. Hub limit was set at 30 during generation of this network visualization.
](./content/figures/fig_lung.png "Tall image"){#fig:lung_vignette height=10in}


## Discussion

In this manuscript, we introduce a new software tool for the analysis and exploration of user data layered on the metabolic reaction network. To improve on tools with similar capabilities, we introduced several new analytical tools and methods to aid the user in the automated identification and discovery of regulatory patterns within their data. These tools and methods include the automated ability to identify regulatory events across reactions, such as a reaction where an input has a high measured abundance and an output has a low measured abundance. Metaboverse also provides dynamic and interactive visualization capabilities to search for patterns and features within the user data manually within classical pathway representations. If a user is interested in how a reaction pattern is propagating across the reaction network and not just a single pathway, they can explore an entity's nearest reaction neighborhood. The user can also explore the connectedness of perturbations across the network and begin to explore hypotheses around the role of connected or disconnected reactions within a particular biological model.

To address the challenge of data sparsity, particularly regarding metabolomics data and the metabolic reaction network, we introduce a reaction collapsing feature. This feature summarizes a series of connected reactions in which values may be missing between the reactions, but where the terminal ends of the reaction path have measured values. Importantly, this ability to collapse reactions augments the capabilities available within Metaboverse, especially in identifying disease-relevant reaction patterns that may be of interest to the user but are otherwise hidden.  

We demonstrated the utility of Metaboverse in exploring single- and multi-omic datasets. We analyzed previously published studies and generated novel analyses highlighting the timecourse and multi-omic capabilities of this tool. We demonstrated that Metaboverse was able to identify regulatory patterns that were expected in the models based on the current literature and identify intriguing patterns that led us to form new hypotheses. We expect these features to be a powerful tool in researchers' toolkits as they analyze their data and plan their next experimental steps. We plan to continue to implement additional features that will expand the applicability of Metaboverse, and have already added several features requested by users since Metaboverse was first released.

Metaboverse aims to enhance the computational toolkit for data analysis and hypothesis generation in metabolic and other experiments. However, numerous challenges remain, which we intend to address in future versions of this software. For example, the reaction collapsing features of Metaboverse aid in identifying patterns across several reactions where data may be missing, but various biological and technical edge cases will need to be accounted for in future implementations of this feature. As more technical limitations in metabolomics data generation and analysis are overcome, we hope that more complete snapshots of metabolism will be visible. Additionally, we currently take a more straightforward and somewhat rudimentary approach to statistical significance integration in the reaction pattern searches. However, more holistic platforms for cross-omics integration are needed and remain a significant challenge across multi-omics research.

In summary, we hope that Metaboverse will bring a new perspective to users' data. We envision Metaboverse will become a staple in the metabolic research toolkit that will help researchers critically and holistically consider their data in the context of biological network interactions and draw the connections needed to extract new and exciting hypotheses that might be challenging without these features.


## Methods
A tutorial for how to use Metaboverse can be found at https://metaboverse.readthedocs.io/en/latest/content/general-usage.html.

### 1. Network curation
Biological networks are curated using the current version of the Reactome database. In particular, the pathway records for each species, complex component and interaction data, Ensembl, and UniProt Reactome mapping tables are integrated into the network database for Metaboverse. Additionally, the ChEBI and The Human Metabolome databases are also referenced for metabolite synonym mapping [@doi:10.1093/nar/gkv1031; @doi:10.1093/nar/gkx1089]. These data are used to generate a series of mapping dictionaries for entities to reactions and reactions to pathways for the curation of the global network.

After the relevant information is parsed from each table or record, the global network is propagated using the NetworkX framework [@url:http://conference.scipy.org/proceedings/SciPy2008/paper_2/full_text.pdf] to generate nodes for each reaction and reaction component and edges connecting components to the appropriate reactions. In some cases, a separate ID is used to generate two nodes for the same metabolite within two separate compartments to aid in downstream visualization; however, user data for the given entity would be properly mapped to both nodes.

After the network is curated for the user-specified organism, each node's degree (or magnitude of edges or connections) is determined to aid in the user's downstream ability to avoid visualizing high-degree components, such as a proton or water, on the metabolic network, which can lead to visual network entanglement and cluttering and a decrease in computational performance [@doi:10.1093/gigascience/giz137].

### 2. Data overlay and broadcasting for missing entities
In order to overlay user data on the global network, first, user-provided gene expression, protein abundance, and/or metabolite abundances' names are mapped to Metaboverse compatible identifiers. For components that Metaboverse is unable to map, a list will be returned to the user so they can provide alternative names to aid in mapping. Second, provided data values are mapped to the appropriate nodes in the network. In cases where gene expression data are available, but protein abundance data is missing, Metaboverse will take the average of the available gene expression values to broadcast to the protein node. For complexes, all available component values (metabolites, proteins, etc.) are averaged. Nodes for which values were inferred will be marked by a dashed border during visualization to clearly show which values are known and which were inferred. Statistical values are derived from the highest value of the components (assuming a scale of 1 denotes no statistical significance and 0 denotes high statistical significance).

### 3. Collapsing reactions with missing expression or abundance values in user data
After data mapping is complete, Metaboverse will generate a collapsed network representation for optional viewing during later visualization. We did, however, choose to enforce a limit of up to three reactions that can be collapsed as data down a pathway should be inferred only so far. We also enforced certain parameters for reaction collapse as follows:   

1. If a reaction has at least one known or inferred value for inputs (substrates) and one known or inferred value for outputs (products), the reaction will be left as is. During the entire reaction collapse step, known catalysts are included when assessing whether a reaction has measured output values (more of a catalyst should lead to more output in most cases), and inhibitors are included when assessing whether the reaction has measured input values (more inhibitor should lead to an accumulation of input in most cases). Catalysts and inhibitors are not included when determining reaction neighbors, as described below.   

2. If a reaction has at least one known input, the input is left as is, and each reaction that shares the same input with the assessed reaction inputs is determined whether it has a measured output. If the neighbor reaction does not contain a known output value, the reaction is left as is. If the neighboring reaction does contain a measured output, the original reaction's inputs and the neighboring reaction's outputs are collapsed to form a single, pseudo-reaction between the two. If the reaction has at least one known output, the inverse is performed where neighbors with components identical to the reaction's inputs are assessed for whether a collapsed reaction can be created.   

3. If a reaction has no measured values, it is determined if the neighboring reactions on both sides (one sharing the reaction's inputs and other sharing the reaction's outputs) have measured values. If both neighbors contain a measured value, a collapsed pseudo-reaction is created, summarizing all three reactions.

For pseudo-reactions, appropriate notes are included to describe the collapse. During visualization, these pseudo-reactions are marked by black dashed edges and dashed node borders. A graphical representation of how this reaction collapse scheme is executed can be found in Figure @fig:collapse_schematic.

![
  **Reaction node collapse schematic.**
  (a) For reactions for which at least one input and at least one output component contain a measured value from the user data, the reaction will be maintained as is. (b) When an input of a reaction is known, but no output has a known value, Metaboverse will search for all neighboring reactions that contain identical inputs. If the neighboring reaction has a known output value, the two reactions will be merged into one pseudo-reaction. (c) When an output of a reaction is known, but no input has a known value, Metaboverse will search for all neighboring reactions that contain identical outputs. If the neighboring reaction has a known input value, the two reactions will be merged into one pseudo-reaction. (d) For reactions with no known values, neighbor pairs that match the inputs and outputs of the considered reaction will be evaluated for whether their respective outputs and inputs both have known values. If values are known for both neighbors, the three reactions will be merged into one pseudo-reaction. (e) As in (d), but if one neighbor does not contain a value and the other does contain a value, no reaction merging will be performed. (f) As in (d), but if neither neighbors contain known values, no reaction merging will be performed. An asterisk (*) indicates the target reaction being considered for a given reaction collapse. A red node indicates a reaction input or output with a measured value. A white node indicates a reaction input or output with no measured value. A gray node indicates a reaction. A gray node with a dashed border indicates a pseudo-reaction. A solid edge indicates a known relationship. A dashed edge indicates a relationship inferred via reaction merging.
](./content/figures/collapse_schematic.png "Tall image"){#fig:collapse_schematic height=6in}

### 4. Regulatory pattern searches and sorting
Metaboverse provides a variety of different regulatory patterns for the user to explore. To identify a reaction pattern is to compare some value that is computed from a reaction or a pathway with a user-specified threshold.

The identified reaction patterns will be listed in a stamp view. Each stamp represents a reaction, with a glyph of the reaction, or the name of the pathway on it. In this stamp view, the identified patterns can be sorted according to three criteria: the number of pathways containing the reaction (not applicable for pathway pattern identification), the magnitude change of the computed value, and the statistical significance. When sorting by the number of pathways or the magnitude change, the identified reactions are arranged in order from the largest to the smallest. When sorting by the statistical significance, reactions with statistical significance on both the input side (substrates) and the output side (products) are listed first by the product of their maximum statistics, followed by the reactions with statistical significance on one of the two sides, and finally the reactions with no statistical significance on both sides. Within each tier, the reactions are sorted from lowest to highest p-values. For all values or statistics used in sorting, only those that determined the reaction pattern are used.

When the stamp view demonstrates draws reaction patterns, all the pathways containing the corresponding reactions will be listed below the stamp view by clicking on the appropriate stamp. For pathway patterns, clicking on a pathway ID will draw the selected pathway as a network, with all the potential regulatory reaction patterns in this pathway highlighted.

### 5. Nearest neighborhood searches and prioritization
To visualize all global connections, a user can select an entity (a gene, protein, or metabolite) and visualize all reactions in which the component is involved. By doing so, the user can visualize other downstream effects a change of one entity might have across the global network, which consequently aids in bridging and identifying any reaction that may occur between canonically annotated pathways. These neighborhoods can be expanded to view multiple downstream reaction steps and their accompanying genes, proteins, and metabolites by modulating the appropriate user option in the app.

The user can also limit which entities are shown by enforcing a degree threshold. By setting this value at 50, for example, the network would not show nodes that have 50 or more connections. One caveat, however, is that this collapsing scheme will occasionally break synchronous pathways into multiple pieces if one of these high-degree nodes were the bridge between two sides of a pathway.

### 6. Perturbation networks
Perturbation networks are generated by searching each reaction in the global network for any reaction where at least one component is significantly perturbed. The user can modify the necessary criteria to base the search on the expression or abundance value or the statistical value and can choose the thresholding value to be used. For the expression thresholding, the provided value is assumed to be the absolute value, so a thresholding value of 3 would include any reactions where at least one component showed a greater than 3 measured change or less than -3 measured change, the value of which is dependent on the data provided by the user thus could represent log<sub>2</sub> fold changes, z-scores, or any other unit appropriate in the biological context.

Once a list of perturbed reactions is collected, the network is constructed, including each of these reactions and their components. Perturbed neighboring reactions that share components are thus connected within the network, and perturbed reactions that are not next to other perturbed reactions are shown as disconnected sub-networks.

### 7. Network visualization and exploration
#### 7.1 Dynamic network plotting
The user interacts with Metaboverse through an interactive app interface. The app employs Electron, a cross-platform app framework that uses JavaScript, HTML, and CSS to design the interface. Metaboverse thus comes packaged as a single executable app with all necessary dependencies included for running on Linux, macOS, and Windows.

Interactive visualization is handled using the D3 and JQuery JavaScript libraries. Force-directed layouts of networks are constructed by taking a user-selected pathway or entity and querying the reactions that are components of the selected pathway or entity. All inputs, outputs, modifiers, and other components of these reactions, along with edges where both source and target are found in the sub-network as nodes, are included and plotted. Relevant metadata, such as user-provided data and reaction descriptions, can be accessed by the user in real-time. Metadata for categorical displays, such as edge or node type, are extracted from the metadata during visualization of the sub-network.

Some performance optimization features are included by default to prevent computational overload. For example, nearest neighbor sub-networks with more than 1500 nodes, or nodes with more than 500 edges, will not be plotted because the plotting of this information in real-time can be quite slow.

#### 7.2 Visualizing pathways and super-pathways
To visualize a pathway, a user selects their pathway of choice, and all component reactions and their substrates, products, modifiers, and metadata are queried from the global network. Super-pathways help categorize these pathways and are defined as any pathway containing more than 200 nodes.

#### 7.3 Visualizing timecourse and multiple condition experiments
Timecourse and multiple condition experiments are automatically detected from the user's input data. When the user provides these data and specifies the appropriate experimental parameters on the variable input page, they will have the option to provide time point or condition labels. Provided data should be listed in the data table in the same order that the labels are provided. Within all visualization modules, the data for each time point or condition can then be displayed using a slider bar, which will allow the user to cycle between time points or conditions.

#### 7.4 Visualizing compartments
Compartments are derived from Reactome annotations. Compartment visualizations are generated using D3's hull plotting feature. Compartment boundaries are defined at the reaction levels and made to encompass each reaction's substrates, products, and modifiers for that given compartment.

#### 7.5 Annotations
Annotations for each reaction are derived from the Reactome database. Pseudo-reactions annotations do not include this information; instead, they include notes on which reactions were collapsed to create the selected pseudo-reaction. All inferred pseudo-reactions and protein or complex values are displayed with dashed edges to differentiate them from measured values.

#### 7.6 Additional features
Metaboverse will continue to undergo development, and new features will be added, but we will briefly highlight some additional features available at the time of publication. We encourage the user to check the documentation for more current updates and information regarding the use of Metaboverse [@url:https://metaboverse.readthedocs.io/en/latest].

##### 7.6a Toggling genes
As gene components can crowd the network visualization space, the user can toggle gene display on and off using the appropriate button. The network is then refreshed to either include or ignore gene components based on their node meta-tag.

##### 7.6b Toggling values
The user can switch between coloring nodes based on the value or statistic provided by toggling the appropriate button. Color bar information for the dataset is saved in the network metadata during curation and used to generate a color bar.

##### 7.6c Toggling features/labels
By default, reaction and feature labels are displayed by hovering the mouse over the node. Reaction or feature nodes can have the labels statically displayed by selecting the appropriate button. An event watch function is used to watch for this user selection and update the display of the node labels.

##### 7.6d Toggling collapsed reactions
By selecting the appropriate button, the user can toggle between displaying a full or collapsed pathway representation of the sub-network. By selecting this button, the network is refreshed using the appropriate reaction dictionary, where for visualization of the collapsed representation, a reaction with available pseudo-reactions substituted for the original reactions is included for network visualization.

##### 7.6e Viewing curated pathway image
Metaboverse visualizes networks dynamically, but the user may be more familiar or comfortable with classical, curated pathway layouts when exploring their data. For a given pathway network, the user can select the appropriate button, and Metaboverse will open a new window with the Reactome curated pathway layout.

##### 7.6f Saving network visualizations
The user can generate a PNG output file for any network created in Metaboverse by selecting the appropriate button.

##### 7.6g Nearest neighbor and hub thresholding
The number of nearest neighbors to visualize, or the limit to the number of edges a visualized node can have, can be modulated by the user using the appropriate input spaces. When visualizing a nearest neighbors network, Metaboverse will recursively fetch related reactions and their neighbors until the critical node display threshold is reached. This searching capability allows the user to visualize downstream effects of a change that may propagate across several reactions. The hub threshold option prevents the plotting of nodes with more than the specified number of edges. This thresholding is handled during the visualization process by excluding any entity nodes that meet these criteria as the neighborhood is propagated. This thresholding is particularly useful in removing hub nodes, such as water or protons, which may be less relevant to the user experience and can quickly clutter the network visualization. This feature can also help plot more extensive neighborhoods, as often neighborhoods quickly link to high-degree nodes, such as water, and limit visualization clarity.

##### 7.6h Metadata display
To help inform the user of selection information and relevant metadata, a space in the legend bar during visualization is reserved for displaying this information, which is updated based on the user's node selections.

### 8. Packaging
The Metaboverse app is packaged using Electron. Back-end network curation and data processing are performed using Python and the NetworkX library. Front-end visualization is performed using Javascript and relies on the D3 and JQuery packages. Saving network representations to a PNG file is performed using the d3-save-svg and string-pixel-width packages (Table @tbl:dependencies). Documentation for Metaboverse is available at https://metaboverse.readthedocs.io. Continuous integration services are performed by Travis CI to routinely run test cases for each change made to the Metaboverse architecture. The Metaboverse source code can be accessed at https://github.com/Metaboverse/metaboverse. The code used to draft and revise this manuscript, as well as all associated scripts used to generate and visualize the data presented in this manuscript can be accessed at https://github.come/Metaboverse/manuscript.

| Name               | Reference                              |
|--------------------|----------------------------------------|
| HTML               | N/A                                    |
| CSS                | N/A                                    |
| Javascript         | N/A                                    |
| Electron           | [@url:https://electronjs.org]      |
| JQuery             | [@url:https://jquery.com]          |
| D3                 | [@url:https://d3js.org]            |
| string-pixel-width | [@url:https://github.com/adambisek/string-pixel-width]               |
| d3-save-svg        | [@url:https://github.com/edeno/d3-save-svg]   |
| Python             | [@url:https://www.python.org/]            |
| pandas             | [@doi:10.5281/zenodo.3509134; @doi:10.25080/Majora-92bf1922-00a]            |
| numpy              |  [@url:https://web.mit.edu/dvp/Public/numpybook.pdf]           |
| scipy              |  [@doi:10.1038/s41592-019-0686-2]           |
| matplotlib         |  [@doi:10.1109/MCSE.2007.55]           |
| NetworkX           |  [@url:http://conference.scipy.org/proceedings/SciPy2008/paper_2/full_text.pdf]           |

Table: Dependencies table. {#tbl:dependencies}

### 9. Validation using biological data

#### 9.1 <i>MCT1</i> perturbation in <i>Saccharomyces cerevisiae</i>

##### Yeast Strains

<i>Saccharomyces cerevisiae</i> BY4743 (MATa/&Alpha;, his3/his3, leu2/leu2, ura3/ura3, met15/MET15, lys2/LYS2) was used to generate the <i>mct1</i>&Delta; strain as described in [@doi:10.1016/j.molcel.2018.06.039].

##### RNA-sequencing sample preparation

RNA sequencing data were generated by growing <i>Saccharomyces cerevisiae</i> biological replicates for strains <i>mct1</i>&Delta; (n=4) and wild-type (n=4). Briefly, cells were grown in glucose and switched to raffinose-supplemented growth medium for 0, 3, and 12 hours such that at the time of harvest, cultures were at OD<sub>600</sub>=1. Cultures were flash-frozen, and later total RNA was isolated using the Direct-zol kit (Zymo Research) with on-column DNase digestion and water elution. Sequencing libraries were prepared by purifying intact poly(A) RNA from total RNA samples (100-500 ng) with oligo(dT) magnetic beads, and stranded mRNA sequencing libraries were prepared as described using the Illumina TruSeq Stranded mRNA Library Preparation Kit (RS-122-2101, RS-122-2102). Purified libraries were qualified on an Agilent Technologies 2200 TapeStation using a D1000 ScreenTape assay (cat# 5067-5582 and 5067-5583). The molarity of adapter-modified molecules was defined by quantitative PCR using the Kapa Biosystems Kapa Library Quant Kit (cat#KK4824). Individual libraries were normalized to 5 nM, and equal volumes were pooled in preparation for Illumina sequence analysis. Sequencing libraries (25 pM) were chemically denatured and applied to an Illumina HiSeq v4 single-read flow cell using an Illumina cBot. Hybridized molecules were clonally amplified and annealed to sequencing primers with reagents from an Illumina HiSeq SR Cluster Kit v4-cBot (GD-401-4001).  Following transfer of the flowcell to an Illumina HiSeq 2500 instrument (HCSv2.2.38 and RTA v1.18.61), a 50 cycle single-read sequence run was performed using HiSeq SBS Kit v4 sequencing reagents (FC-401-4002).

##### Sequence analysis

Sequence FASTQ files were processed using XPRESSpipe [@doi:10.1371/journal.pcbi.1007625]. Batch and log files are available at [@url:https://github.com/Metaboverse/manuscript/tree/master/data/sce_mct1_omics]. Notably, reads were trimmed of adapters (AGATCGGAAGAGCACACGTCTGAACTCCAGTCA). Based on library complexity quality control, de-duplicated alignments were used for read quantification due to the high number of duplicated sequences in each library. Differential expression analysis was performed using DESeq2 [@doi:10.1186/s13059-014-0550-8] by comparing <i>mct1</i>&Delta; samples with wild-type samples at the 12-hour time-point to match the steady-state proteomics data. log<sub>2</sub>(fold change) and false discovery rate ("p-adj") values were extracted from the DESeq2 output.

##### Proteomics analysis

Steady-state quantitative proteomics data were generated as described in [@doi:10.1016/j.molcel.2018.06.039]. Briefly, cells were grown in glucose and switched to raffinose-supplemented growth medium overnight and harvested at mid-log phase. For this analysis, we compared the <i>mct1</i>&Delta; (n=3) with the wild-type (n=3) cell populations. log<sub>2</sub>(fold change) values and Benjamini-Hochberg corrected p-values were generated by comparing <i>mct1</i>&Delta; with the wild-type cells. P-values were generated before correction using a 2-tailed, heteroscedastic Student's T-test.

##### Metabolomics sample preparation

Metabolomics data were generated by growing the appropriate yeast strains in synthetic minimal media (S-min) supplemented with 2% glucose until they reached OD<sub>600</sub> between 0.6-0.8 (n=6; except in one 3-hour wild-type sample, where n=5). Cells were then transferred to S-min media containing 2% raffinose and harvested after 0, 15, 30, 60, and 180 minutes (n=6/time-point/strain, except for the 3-hour wild-type samples, where n=5).

##### Metabolite extraction

A 75% boiling ethanol (EtOH) solution containing the internal standard d4-succinic acid (Sigma 293075) was then added to each sample. Boiling samples were vortexed and incubated at 90 °C for 5 minutes. Samples were then incubated at -20 ˚C for 1 hour. After incubation, samples were centrifuged at 5,000 x g for 10 minutes at 4˚C. The supernatant was then transferred from each sample tube into a labeled, fresh 13x100mm glass culture tube. A second standard was then added (d27-myristic acid CDN Isotopes: D-1711). Pooled quality control samples were made by removing a fraction of collected supernatant from each sample, and process blanks were made using only extraction solvent and no cell culture. The samples were then dried <i>en vacuo</i>. This process was completed in three separate batches.

##### Mass spectrometry analysis of samples

All GC-MS analysis was performed with an Agilent 5977b GC-MS MSD-HES and an Agilent 7693A automatic liquid sampler. Dried samples were suspended in 40 µL of a 40 mg/mL O-methoxylamine hydrochloride (MOX) (MP Bio #155405) in dry pyridine (EMD Millipore #PX2012-7) and incubated for 1 hour at 37 °C in a sand bath. 25 µL of this solution were added to auto sampler vials. 60 µL of N-methyl-N-trimethylsilyltrifluoracetamide (MSTFA with 1%TMCS, Thermo #TS48913) were added automatically via the auto sampler and incubated for 30 minutes at 37 °C. After incubation, samples were vortexed, and 1 µL of the prepared sample was injected into the gas chromatograph inlet in the split mode with the inlet temperature held at 250 °C. A 10:1 split ratio was used for the analysis of the majority of metabolites. For those metabolites that saturated the instrument at the 10:1 split concentration, a split of 50:1 was used for the analysis. The gas chromatograph had an initial temperature of 60 °C for 1 minute followed by a 10 °C/min ramp to 325 °C and a hold time of 5 minutes. A 30-meter Phenomenex Zebron AB-5HT with 5m inert Guardian capillary column was employed for chromatographic separation. Helium was used as the carrier gas at a rate of 1 mL/min.

##### Analysis of mass spectrometry data

Data were collected using MassHunter software (Agilent). Metabolites were identified, and their peak area was recorded using MassHunter Quant. These data were transferred to an Excel spreadsheet (Microsoft, Redmond WA). Metabolite identity was established using a combination of an in-house metabolite library developed using pure purchased standards, the NIST and Fiehn libraries. Resulting data from all samples were normalized to the internal standard d4-succinate. P-values were derived using a heteroscedastic, two-tailed Student's T-test and adjusted using the Benjamini-Hochberg correction procedure.

#### 9.2 Human lung adenocarcinoma metabolomics
Data were accessed from Metabolomics Workbench [@doi:10.21228/M8PG66] and processed as in our previous re-study of these data [@doi:10.1093/gigascience/giz137]. P-values were derived using a paired, two-tailed Student's T-test and adjusted using the Benjamini-Hochberg correction procedure.

### 10. Data availability

<i>mct1</i>&Delta; and accompanying wild-type transcriptomics timecourse data is deposited at the GEO repository under identifier GSE151606 [@url:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE151606]. Metabolomics data is deposited at the Metabolomics Workbench repository under project identifier PR000961, study identifier ST001401 [@doi:10.21228/M8GD71].

The curated networks for these data are available at [@url:https://github.com/Metaboverse/manuscript/tree/master/data/sce_mct1]. Networks were generated by taking the 12-hour transcriptomics and proteomics datasets with their appropriate log<sub>2</sub>(fold change) and statistical values, along with the 0, 15, 30, 60, and 180 minute metabolomics datasets with their respective log<sub>2</sub>(fold change) and statistical values and layering these data on the <i>Saccharomyces cereviseae</i> global reaction network as curated by Metaboverse from the Reactome database. Reaction patterns and perturbation network analyses were performed within the Metaboverse platform.

The Metaboverse source code is available at https://github.com/Metaboverse/Metaboverse. The latest version of the software can be found at https://github.com/Metaboverse/Metaboverse/releases/latest.

The source code and data for this manuscript and the subsequent analyses are available at [@url:https://github.com/Metaboverse/manuscript].


## Acknowledgements

We thank Thomas O'Connell for valuable feedback and feature suggestions during beta-testing of the software. We thank Alex J. Bott (U. of Utah), Ahmad A. Cluntun (U. of Utah), Kevin G. Hicks (U. of Utah), Jeffrey T. Morgan (U. of Utah), and other members of the Rutter lab for their thoughtful insights and suggestions. We thank Brian Dalley and the University of Utah High-Throughput Genomics Core for help with RNA library preparation and sequencing. We also thank Christine Pickett for providing editing services for this manuscript. The support and resources from the Center for High-Performance Computing at the University of Utah are gratefully acknowledged.

## Funding

J.A.B. is supported by the National Cancer Institute (NCI) of the National Institutes of Health (NIH) under Award Number 1F99CA253744 (https://www.cancer.gov/) and received additional support from the National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK) Inter-disciplinary Training Grant T32 Program in Computational Approaches to Diabetes and Metabolism Research, 1T32DK11096601 to Wendy W. Chapman and Simon J. Fisher (https://www.niddk.nih.gov/). S.M.N. received support from The United Mitochondrial Disease Foundation PF-15-046 (https://www.umdf.org/) and American Cancer Society PF-18-106-01 (https://www.cancer.org/) postdoctoral fellowships, along with T32HL007576. J.E.C. is funded by S10OD016232, S10OD021505, and U54DK110858. This work was supported by funds from NIDDK fellowship 1T32DK11096601 and NCI fellowship 1F99CA253744 (to J.A.B.) (https://www.niddk.nih.gov/, https://www.cancer.gov/), (NSF grants NSF DBI-1661375 and NSF IIS-1513616 (to B.W.) (https://www.nsf.gov/), and NIH grant R35GM131854 (to J.R.) (https://www.nih.gov/). The computational resources used were partially funded by the NIH Shared Instrumentation Grant 1S10OD021644-01A1 (https://www.nih.gov/). Mass spectrometry equipment was obtained through NCRR Shared Instrumentation Grant 1S10OD016232-01, 1S10OD018210-01A1, and 1S10OD021505-01 (to J.E.C.). The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.


## Contributions

* Conceptualization: J.A.B., T.C.W., B.W., J.R.
* Supervision: J.A.B., B.W., J.R.
* Project Administration: J.A.B.
* Investigation: J.A.B., T.C.W., Y.O., S.M.N., T.V.
* Formal Analysis: J.A.B.
* Software: J.A.B., Y.Z.
* Methodology: J.A.B., Y.Z.
* Validation: J.A.B., Y.O., I.G.
* Data Curation: J.A.B., T.C.W.
* Resources: J.A.B., J.E.C., B.W., J.R.
* Funding Acquisition: J.A.B., J.E.C., B.W., J.R.
* Writing - Original Draft Preparation: J.A.B.
* Writing - Review & Editing: J.A.B., Y.Z., T.C.W., Y.O., S.N.M., J.E.C., B.W., J.R.
* Visualization: J.A.B., Y.O.


## Supplementary Material {.page_break_before}

### Reaction collapsing reveals multi-reaction putative regulatory patterns
Yeast timecourse metabolomics processed data were obtained as measured and discussed in this study in Vignette 1. Metaboverse reaction pattern searches identified a pattern across the two step transformation of glutathione and methylglyoxal into lactate at 60 minutes after carbon source shift. As the intermediate metabolite was not measured in the dataset, but the inputs to the first reaction and the outputs to the second reaction were measured, Metaboverse collapsed the two reactions into one, thus enabling pattern recognition across two reactions (Figure @fig:supp_fig_collapse). Conventional methods might have missed such a reaction without summarizing intermediate steps where the intermediate metabolite(s) are not measured. This transformation is particularly important in this model as lactate is closely tied with TCA metabolism and may provide explanatory mechanisms for the changes in TCA metabolism observed in this model.

![
  **Reaction collapsing reveals putative regulatory patterns across TCA metabolism-proximal reactions in mtFAS-deficient yeast.**
  (Left) Metabolomics values are shown as node shading, where an increasingly blue shade indicates down-regulation, and an increasingly red shade indicates up-regulation. Measured log<sub>2</sub>(fold change) and statistical values for each entity are displayed below the node name. A gray node indicates a reaction. A bold gray node with a purple border indicates a pattern was found at this reaction for the given data type time points. A dashed purple border indicates a pattern was discovered across collapsed reactions. Circles indicate metabolites, squares indicate protein complexes, diamonds indicate proteins, and triangles indicate genes. Gray edges are core relationships between reaction inputs and outputs. Green edges indicate a catalyst. Dashed blue edges point from a metabolite component to the complex it is involved in. Dashed orange edges point from a protein component to the complex it is involved in. The background shading indicates different cellular compartments for each color. The displayed values are for the 60 minute timepoint. (Right) A screenshot from the Reactome (v73; https://reactome.org/PathwayBrowser/#/R-HSA-1428517&PATH=R-HSA-1430728) pathway explorer of this two-step reaction set.
](./content/figures/supp_fig_collapse.png "Square image"){#fig:supp_fig_collapse}


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>
