## Introduction

Metabolism is a complex network of reactions and interactions between genes, enzymes, complexes, and metabolites [@doi:10.3390/metabo9040076]. To understand these complex components, researchers normally adopt a reductionist approach to teasing apart the characteristics and mechanics of these processes and how they fit into the larger picture of biology and disease. While a vital component in the scientific process, by doing so, many interesting properties of metabolism can be missed. For example, in differential gene expression analysis, researchers rely on thresholds of magnitude and statistical significance to prioritize genes for follow-up study. However, this can inadvertently limit the scope of study of metabolism when in fact metabolism is a highly interconnected system where distal components and their modulation can have rippling effects across the network. The current approach is analogous to telling the story of Little Red Riding Hood, but only by reading the 20 most frequent words used in the study. Certainly doing so efficiently highlights key words like "wolf" and "little red riding hood," but also prevents a coherent story from being told and would make it difficult for someone who had never heard the story of Little Red Riding Hood from comprehending the actual plot.

Over the past decade, several computational tools have emerged and become popular for their focus on trying to solve these issues in data contextualization. We will highlight four, and while others exist, we focus on tools representative and most popular for their respective properties. First is MetaboAnalyst, which relies largely on set enrichment methods for analysis of data, or looking at the belongingness of sets of significantly changed analytes (i.e. metabolite, protein, or gene measurements), for extracting interesting information. While network visualization is available, it focuses primarily on interaction networks and its ability to extract regulatory information is limited, particularly in an automated fashion [@doi:10.1093/nar/gkp356; @doi:10.1093/nar/gky310]. Second is Cytoscape, which serves as a general go-to platform for representing biological or other networks. One strength of Cytoscape is its ability for users to design their own apps, or plug-ins to develop customized analyses. However, comprehensive regulatory identification methods are unavailable [@doi:10.1101/gr.1239303]. MetExplore focuses on the curation of networks, and is particularly useful for collaborative annotation of emerging organisms. It additionally can layer experimental data on the network for visualization [@doi:10.1093/nar/gky301; @doi:10.1093/nar/gkq312]. Reactome, which Metaboverse uses for the curation of biological networks, also offers analytical tools for user data, but again relies on set enrichment or manual methods for identifying patterns. While all have their respective utility, there is still a need for tools that automate pattern and trend detection, especially when data is sparse, across metabolic networks in order to extract regulatory and other features from data [@doi:10.1093/nar/gki072; @pmid:31691815; @pmid:29145629].

In order to address these limitations in current conventions of metabolic data analysis, contextualization, and interpretation, we created the operating-system-independent desktop application, Metaboverse, to aid users in filling in and expanding upon the details of their model's metabolic story. Metaboverse is an interactive tool for exploratory data analysis that searches user data in the context of the metabolic network to identify interesting patterns and trends within the data. Metaboverse will aid scientists in formulating new hypotheses from their data and aid them in designing follow-up experiments for a deeper understanding of their model. Metaboverse operates across the entire metabolic network to quickly and automatically detect patterns and trends from a pre-designed pattern library, or can accept interactive input from the user where they can define certain patterns or trends they would like to identify across the global metabolic network. Figure @fig:overview provides a graphical abstract to illustrate <i>Metaboverse's</i> role in the exploratory data analysis of biological data in the context of metabolism.

![
  **Metaboverse conceptual overview.**
  Illustration summarizing the usage of Metaboverse to use biological data layered on the global reaction network to rapidly identify regulatory hotspots. Traditionally, when a scientist performs an -omics experiment, they tend to focus on a couple of features that are differentially regulated. Metaboverse contextualizes the data across the metabolic network and identifies interesting regulatory patterns in the data.
](./content/figures/graphical_abstract.png "Square image"){#fig:overview}

In order to provide a platform for the exploration of single or multi-omic metabolic data, we developed several computational features to aid in the aims discussed above. First, we developed a pattern search engine for the rapid and automated identification of patterns and trends in -omic data on the metabolic network. Conceptually, this search engine borrows principles of topological motif searching from graph theory [@doi:10.1126/science.1089167; @doi:10.1038/ng881] and builds upon the concept of "activity motifs" [@doi:10.1038/nbt.1499]. Another feature introduced in Metaboverse allows for the interactive exploration of specific reactions or reaction entities with on-the-fly pattern search analysis. Users can explore specific pathways of interest and look for other interesting patterns and trends in pathways of interest. Perturbations in the abundance or behavior of a particular metabolic network component can lead to downstream genotypic and phenotypic modulations in a biological system. We include an interactive perturbation connectivity module to allow users to explore the effects of perturbations on their system. Finally, we tackle the challenge of sparsity, particularly in metabolomics datasets, by introducing a reaction collapsing or bridging feature that allows for searching for motifs across multiple reactions, where intermediate steps of the reaction trajectory may not have measured values for their respective input or output components.

Metaboverse is designed to handle standard two-condition experiments, flux metabolomics (in progress), and time-course or multi-condition experiments. Users can provide any combination of datasets from each -omics categories (values mapping to gene, protein, and metabolite identifiers). Time-course inputs can consist of a matched samples for every time point, or can provide a combination of steady-state levels and time points. Users input fold change and appropriate statistical measures with proper multiple-testing adjustments from their respective -omics datasets, which Metaboverse then layers onto the metabolic network. Metaboverse can handle data from a variety of model organisms, including humans, mouse, yeast, zebrafish, and more. The foundational curation of Metaboverse is built on Reactome reaction curations, thus providing 90+ species which the Metaboverse environment can process. In order to validate these methodologies available in Metaboverse we analyzed a variety of datasets to provide representative vignettes that highlight <i>Metaboverse's</i> reliability in extracting canonical features, as well as utility in identifying novel features and patterns. We expect that Metaboverse will become a foundational piece of the analytical toolkit and augment one's ability to more deeply and holistically explore metabolism and other reaction networks and aid in our ability to provide clearer context within these models.
