## Introduction

Metabolism is a complex network of reactions and interactions between genes, enzymes, protein complexes, and metabolites [@doi:10.3390/metabo9040076]. To understand these complex components, researchers frequently adopt reductionist approaches to tease apart the characteristics and mechanics of these processes and determine how they fit into the larger picture of biology and disease. Such approaches are a vital component in the scientific process, but a reductionist approach may miss many essential properties of metabolism. For example, in differential gene expression analysis, researchers rely on thresholds of magnitude and statistical significance to prioritize genes for follow-up study. However, this can inadvertently limit the scope of study of metabolism when, in fact, metabolism is a highly interconnected system where distal components and their modulation can have coordinated or rippling effects across the network [@doi:10.1016/j.coisb.2018.10.001; @doi:10.1038/s41540-018-0067-y]. Biological perturbations involve complex, cooperative effects, many of which may seem negligible in isolation. The current approach is analogous to telling the story of Little Red Riding Hood by reading only the 20 most frequent words. Indeed, doing so efficiently highlights keywords like "wolf" and "little red riding hood," but also prevents a coherent story from being told. This approach makes it difficult for someone who has never heard the story of Little Red Riding Hood to comprehend the plot and especially the higher-level interpretations of the story.

While several computational tools have risen to prominence over the past decade to try to resolve these issues in data contextualization, all lack the ability to rapidly extract a comprehensive list of regulatory patterns within user data [@doi:10.1093/nar/gkp356; @doi:10.1093/nar/gky310; @doi:10.1101/gr.1239303; @doi:10.1093/bioinformatics/btx012; @doi:10.1093/bioinformatics/btr661; @doi:10.1093/bioinformatics/btq048; @doi:10.1093/nar/gky301; @doi:10.1093/nar/gkq312; @doi:10.1093/bioinformatics/btx588; @doi:10.1093/nar/gki072; @pmid:31691815; @pmid:29145629] (a brief discussion of these referenced representative tools for metabolic network analysis is available in the Supplementary Methods). This inability to extract vital information from the metabolic network is particularly pronounced in cases where experimental data have sparse coverage of the metabolic network, a common situation in current metabolomics datasets. To address these limitations in metabolic data analysis, contextualization, and interpretation, we created Metaboverse, a cross-platform, interactive application to aid users in exploring and interpreting their model's metabolic story and generating new hypotheses from the available information (Figure @fig:overview). Importantly, Metaboverse introduces interactive modules that enable the rapid and automated discovery of regulatory patterns across all reactions in the network and allows for the identification of these patterns across sets of multiple reactions. We also incorporate modules that allow for the exploration of reaction neighborhood perturbations and super-pathway total perturbation networks. Since the initial public release of Metaboverse, we have also received feedback from multiple users. These requested features are also available within Metaboverse.

![
  **Metaboverse rapidly identifies potential regulatory patterns within the metabolic network using user data.**
  Illustration summarizing the usage of Metaboverse to model biological data on the reaction network to rapidly identify regulatory patterns. Traditionally, when scientists perform an omics experiment, they tend to focus on a few features that are differentially regulated. Metaboverse allows a user to input multiple omics data types, which it layers upon the metabolic network. Metaboverse then uses this integrated model to identify patterns of putative regulatory potential within the data. The user can also dynamically explore metabolic pathways and other network representations.
](./content/figures/graphical_abstract.png "Tall image"){#fig:overview height=3.5in}

To demonstrate the ability of Metaboverse to identify canonical, expected, and novel patterns within the reaction networks and showcase its other interactive analysis features, we analyzed single- and multi-omics datasets of multiple organisms. With the ranked output of Metaboverse's search pattern algorithm, we are able to formulate new hypotheses for each model. We expect that Metaboverse will become a foundational tool in the analytical toolkit and augment the user's ability to more holistically explore and contextualize metabolic reaction networks.
