## Introduction

Metabolism is a complex network of reactions and interactions between genes, enzymes, protein complexes, and metabolites [@doi:10.3390/metabo9040076]. To understand these complex components, researchers frequently adopt a reductionist approach to tease apart the characteristics and mechanics of these processes and see how they fit into the larger picture of biology and disease. Such approaches are a vital component in the scientific process, but a reductionist may miss many essential properties of metabolism. For example, in differential gene expression analysis, researchers rely on thresholds of magnitude and statistical significance to prioritize genes for follow-up study. However, this can inadvertently limit the scope of study of metabolism when, in fact, metabolism is a highly interconnected system where distal components and their modulation can have rippling effects across the network [@doi:10.1016/j.coisb.2018.10.001; @doi:10.1038/s41540-018-0067-y]. The current approach is analogous to telling the story of Little Red Riding Hood by reading only the 20 most frequent words. Indeed, doing so efficiently highlights keywords like "wolf" and "little red riding hood," but also prevents a coherent story from being told and makes it difficult for someone who had never heard the story of Little Red Riding Hood to comprehend the plot (Figure @fig:little_red). In short, biological perturbations involve complex, cooperative effects, many of which seem negligible in isolation.

![
  **Word cloud of 20 most frequent words or phrases in the story of Little Red Riding Hood.**
  Text taken from the original telling of Little Red Riding Hood by Charles Perrault. Words and phrases derived using [@url:https://github.com/amueller/word_cloud]. Outline derived from the piece, "Little Red Riding Hood" by J.W. Smith.
](./content/figures/red_riding.png "Tall image"){#fig:little_red height=4in}

Over the past decade, several computational tools have risen to prominence for their attempts to resolve these issues in data contextualization. We will highlight four such tools that are popular and representative for their respective properties. First is MetaboAnalyst, which relies heavily on set enrichment methods for analysis of data, or examining the belongingness of sets of significantly changed analytes (i.e., metabolite, protein, or gene measurements), for extracting interesting information. Network visualization is available, but it focuses primarily on interaction networks, and its ability to extract regulatory information is limited, particularly in an automated fashion [@doi:10.1093/nar/gkp356; @doi:10.1093/nar/gky310]. Second is Cytoscape, which serves as a general go-to platform for representing biological or other networks. One strength of Cytoscape is the ability to design apps or plug-ins to develop customized analyses; however, comprehensive and metabolism-specific regulatory identification methods are unavailable [@doi:10.1101/gr.1239303]. One plug-in for Cytoscape that is focused on metabolic data is MetScape, but again, this tool is limited to pathway enrichment, correlation networks, and simple data visualization and does not integrate approaches to identify regulatory mechanisms within the data [@doi:10.1093/bioinformatics/btx012; @doi:10.1093/bioinformatics/btr661; @doi:10.1093/bioinformatics/btq048]. MetExplore focuses on the curation of networks and is particularly useful for collaborative annotation of emerging models of organisms with incomplete metabolic network curations. It additionally can layer experimental data on the network for visualization [@doi:10.1093/nar/gky301; @doi:10.1093/nar/gkq312]. Reactome, which Metaboverse uses for the curation of biological networks, also offers analytical tools for user data, but again relies on set enrichment or manual methods for identifying patterns. While all have their respective utility, there is still a need for tools that integrate these features and automate pattern and trend detection across metabolic networks in order to extract regulatory and other features from data [@doi:10.1093/nar/gki072; @pmid:31691815; @pmid:29145629]. This need is particularly pronounced in common cases where experimental data have sparse coverage of the network, a situation common in modern metabolomics.

To address these limitations in current conventions of metabolic data analysis, contextualization, and interpretation, we created Metaboverse, an operating-system-independent application, to aid the user in understanding and interpreting their model's metabolic story. Metaboverse is an interactive tool for exploratory data analysis that searches user data in the context of the metabolic network to identify unusual patterns and trends within the data. By identifying these trends and patterns across the global reaction network, Metaboverse will aid scientists in formulating new hypotheses from their data and in designing follow-up experiments for a deeper understanding of their model. In addition to a pre-designed pattern search library, additional patterns will be included in future releases of Metaboverse, along with the ability for a user to interactively design patterns of interest to be included in the pattern search library.

To provide a platform for the exploration of single or multi-omic metabolic data, we developed several computational features to aid in addressing the aims discussed above. We developed a pattern search engine for the rapid and automated identification of patterns and trends in omics data on the metabolic network. Conceptually, this search engine borrows principles of topological motif searching from graph theory [@doi:10.1126/science.1089167; @doi:10.1038/ng881] and builds upon the concept of "activity motifs" [@doi:10.1038/nbt.1499]. We will use "search pattern" and "motif" interchangeably. Another feature introduced in Metaboverse allows for the interactive exploration of specific reactions or reaction entities with on-the-fly pattern search analysis. The user can explore specific pathways of interest and look for other interesting patterns and trends within these pathways. Perturbations in the abundance or behavior of a particular metabolic component can lead to downstream gene expression changes and other phenotypic modulations in a biological system. We include an interactive perturbation connectivity module to allow the user to explore the effects of perturbations on their system.

Finally, we tackle the challenge of sparsity, particularly in metabolomics datasets. We designed a reaction collapsing feature that allows for searching for motifs across multiple reactions, where intermediate steps of the reaction trajectory may not have measured values for their respective input or output components. This feature is particularly important in allowing for pattern detection across missing components of a pathway. For example, the human metabolic network alone consists of over 1000 metabolite nodes, but an average gas chromatography mass spectrometry (GC-MS) experiment may quantify only 100-200 metabolites by name. By bridging proximal reactions, one can detect patterns that would require the comparison of the terminal ends of that sub-pathway. This reaction collapsing feature can help the user identify more inconspicuous patterns within the global reaction network.

Metaboverse is currently designed to handle standard two-condition experiments and timecourse or multi-condition experiments. The user can provide any combination of datasets from each omics category (values mapping to gene, protein, and metabolite identifiers). Timecourse inputs can consist of matched samples for every time point or can provide a combination of steady-state levels and time points. The user inputs fold change and appropriate statistical measures with proper multiple-testing adjustments from their respective omics datasets, which Metaboverse then layers onto the metabolic network. Metaboverse is designed to accept data from a variety of model organisms, including humans, mice, yeast, and zebrafish. The foundational curation of Metaboverse is built on Reactome reaction curations, providing 90+ species that the Metaboverse environment can process. To validate these methodologies available in Metaboverse, we analyzed a variety of datasets to provide representative vignettes that highlight the reliability of Metaboverse for extracting canonical features, as well as its power for identifying novel features and patterns. We expect that Metaboverse will become a foundational piece of the analytical toolkit and augment the user's ability to more holistically explore metabolism and other reaction networks within a clearer context.
