## Results

### Metaboverse is a dynamic, user-friendly tool for the exploration of high-throughput biological data in organism-specific pathways

#### Overview
We designed Metaboverse as a lightweight, self-contained app for the dynamic exploration of high-throughput biological data. The pathway curations are derived from Reactome [@doi:10.1093/nar/gkz1031], coupled with metabolite name cross-referencing from the ChEBI [@doi:10.1093/nar/gkv1031] and the Human Metabolome (HMDB) [@doi:10.1093/nar/gkx1089] databases. As of the time of writing this manuscript, Metaboverse is capable of analyzing data for over 90 species. A user begins by providing the desired output location for a new curation and organism of interest. If the user has previously curated data onto a reaction network for the organism of interest, they can alternatively provide this intermediate file. Next, the user provides the relevant transcriptomic, proteomic, and/or metabolomic datasets to layer onto the global reaction network of their organism of interest. Input data types can be extended to any dataset that uses the relevant mapping IDs; for example, one could provide ribosome profiling translation efficiency values mapped to the appropriate gene IDs for layering onto the network and downstream analysis. For each omics type, the user provides fold change and statistical values for each measurement. For example, if a user provided data for a two-condition comparison transcriptomics dataset, these values would consist of log<sub>2</sub> Fold Change values between the experimental and wild-type conditions, as well as the appropriate statistical values. Transcriptomics count data follows a negative binomial distribution, so the appropriate adjusted p-values based on multiple hypothesis testing should be used [@doi:10.1186/s13059-014-0550-8, @doi:10.1093/bioinformatics/btp616]. For other omics types, these multiple hypothesis testing methods will vary based on the data distribution and other conventions. During the data input step, the user will also specify a few experimental parameters for consideration during downstream analysis and visualization. Following these user inputs, the organism's network, data processing, and motif analysis (discussed further below) are curated, and a curation file is output for future analysis (see Figure @fig:package_overview).

We chose to format network visualizations in a reaction-centric format, which means that a sub-network unit is comprised of a reaction node, linked with its reaction input and reaction output nodes, along with any modifier nodes. If modifier nodes are a protein complex, the component proteins of that complex are linked to the complex node. Any genes available in the database are linked to their related proteins. Each node and link-type are color-coded to help the user quickly distinguish the details of a reaction sub-network structure. Reactions with matching input or output sides are then naturally linked, allowing a multi-reaction pathway to emerge from this network structure. This formatting differs from other formatting styles the user may be familiar with, such as KEGG pathway maps, for various of reasons. This methodology dynamic visualization differs from more classical methods, so each pathway in the visualization module contains a button linking to a more classically formatted representation of the pathway for reference. While classical pathway definitions are used to group sets of reactions during visualization, analytical methods such as the motif search in Metaboverse search the global reaction network. This approach aids in making pattern analysis within the reaction network less subject to biases arising from constraining analysis to a pathway that is not directly contextualized with its place in the global network.

![
  **Metaboverse package overview.**
  Illustration outlining the data input, processing, and exploration steps of the Metaboverse package. The user provides the name of the organism of interest from a drop-down menu along with an output location. The user then has the option to provide transcriptomics, proteomics, and or metabolomics datasets. These datasets can be single- or multi-condition or timecourse experiments. Data is formatted as follows: row names are the measured entity names or IDs, the first column is log<sub>2</sub> Fold Change or other measurement values, and the second column is statistical measurements. During this step, the user can also provide sample labels and other modifiers that will customize the curation of the data on the reaction network. Metaboverse will then build the model. Once the model is complete, the user will be able to visualize the reaction motifs identified, explore pathway or global perturbation networks, or perform general pathway exploration of the data.
](./content/figures/package_overview.png "Square image"){#fig:package_overview}

#### Handling data sparsity within the biological global network
Missing data points, particularly in metabolomics experiments, are frequent and can make the analysis of pathways and identification of regulatory patterns in the network challenging. Thousands of metabolites are known to participate in human metabolism, but the current state of the technology used for determining which mass spectra belong to which metabolite can be challenging and often results in a limited number of data points being quantified. This limitation can lead to gaps in the metabolic network, which can be challenging to explore and analyze. For example, the human metabolic network is comprised of over 1000 metabolites, whereas the average GC-MS experiment may measure only 100-200 metabolites by name, leading to measurement gaps between reactions in a pathway. We, therefore, developed a reaction compression algorithm (detailed more in the Methods section) that collapses up to three reactions with missing data points if they can be bridged with known data on the distal ends of the reaction path. Similar concepts have been used in metabolomic analysis before to identify amino acid-related metabolites [@doi:10.1038/s41586-019-1437-3], but we introduce the first computational implementation to our knowledge and adopt a slightly more conservative collapsing scheme. These reactions, or pseudo-reactions, are visually distinct to show the user where reactions were collapsed and understand what that pseudo-reaction summarizes.

[comment]: <> (We found that these collapsed reaction representations aided significantly in identifying intriguing regulatory mechanisms known to be active in lung tumors, but which would have been missed by traditional, uncollapsed representations of human metabolism (see Figure @fig:supp_fig_collapse and accompanying supplementary text for further discussion).)

#### Rapid identification of interesting regulatory patterns in the reaction network
Following network curation, the user can visualize available reaction motifs identified across the global reaction network. In a computational science context, a motif is a recurring pattern in network structure or the organization of network entities and their relationships to one another. However, with omic data, we are more interested in identifying patterns in expression or abundance of genes, proteins, and metabolites. Previous work by Checkik, et al. introduced the concept of "activity motifs," where instead of identifying motifs based on network structure, they were identified using the expression characteristics of nodes in transcription factor binding signaling networks [@doi:10.1038/nbt.1499]. We adapted this methodology to identify and interactively display interesting regulatory hotspots within the global reaction network. For example, a reaction's inputs may exhibit increased levels and its outputs decreased levels, indicating a regulatory event occurring at the reaction.

In Metaboverse, we define a motif as a regulatory pattern identified across a reaction or pseudo-reaction. Metaboverse contains a library of default motifs to search for within the global reaction network, and the user can define custom motifs they would like to identify across the global network (please refer to the documentation for more details on available motifs [@url:https://metaboverse.readthedocs.io/en/latest]). Metaboverse will search the global network from a pre-defined library of regulatory patterns (regulatory hotspots, or motifs) and return a graphical stamp view of conserved patterns. These motifs by default are ordered by the statistical values associated with the entity components of the motif, but can be ordered by the strength of the motif based on the magnitude of change. Statistical values associated with each measurement for a measured entity as provided by the user are used to weight these motifs and prioritize returned results. We use a three-tiered sorting strategy when sorting by the associated statistical values. The highest prioritized motifs are those where the relevant components on either side of the reaction (inputs vs. outputs) that determined the reaction motif are statistically significant. Of these motifs from the first tier, motifs are sorted by lowest to highest cumulative p-value or other relevant statistical value. In the second sorting tier, reactions with at least one side of the reaction with a statistically significant motif component are sorted by statistical strength. Finally, all other motifs are sorted by the cumulative statistical value for the motif components.

Reaction motif nodes are visually distinct to quickly draw the user's attention to these interesting patterns. For a given pattern, the user can then explore each pathway in which this particular motif is found. Motif analysis of the global regulatory network will allow the user to rapidly identify interesting features within the data, particularly patterns between canonical pathways or in other pathways that may appear tangential in their research, but are instead rather striking based on these contextualized analytical methods. In the data vignettes below, we demonstrate this utility further.

[comment]: <> (The user will also be able to design their own patterns through an interactive pattern drawing tool and even design specific scenarios that are cognizant of feature type (in progress). For example, one might be interested in a pattern where a protein displays higher expression, but the resulting metabolite is decreased. This pattern could then be coded interactively into the Metaboverse framework by the user.)

#### Dynamic visualization of organism-specific reaction pathways
Following the curation of the global network as described above, the user can manually search individual canonical pathways or individual entities and their reaction neighborhoods. For a given selection, all relevant reactions that are annotated as a part of that pathway will be visualized, along with their core input (substrate) and output (product) components. In addition to these core elements, known catalysts and inhibitors are included, as well as the component proteins, genes, and metabolites that form functional complexes involved in a particular reaction. For complex nodes, simulated values are calculated by taking the average of all measured component entities in that complex. Assuming a statistical value between 0 and 1, simulated statistics are calculated by taking the maximum statistic value of all measured component entities in that complex. In cases where a gene value is known, but its protein value is unmeasured, the protein value can be inferred using aggregated gene component values. Relevant pathway and analytical metadata are also displayed. All reaction motif types are displayed using default thresholds if present in a selected pathway. Additional aids for visualization are also available, such as the ability to remove nodes from the visualization that contain a high number of relationships to other network features so that these nodes, which act as hubs in the network, do not lead to cluttered representations. Often, these hub nodes are ubiquitous features such as water and protons, which may be of limited interest to the user during data visualization [@doi:10.1093/gigascience/giz137]. Compartment domains are also visualized to include relevant reactions and their components that occur in a given cellular compartment.

#### Visualization of downstream effects of network perturbations using reaction neighborhoods
The user may be interested in a particular metabolite or protein and the downstream effects their perturbation has on related pathways. By double-clicking a node of interest, or by selecting the entity name from the drop-down menu, the user can explore all downstream effects across all pathways in the global network. The user can also define how many neighborhoods to display such that two or more reaction steps downstream or upstream of the selected entity can be visualized [@doi:10.1007/PL00009293; @doi:10/bz69nr]. This functionality moves the analysis away from traditional, pathway-centric approaches, and helps contextualize the far-reaching effects changes in metabolism or other biological systems can have across classical pathways.

#### Exploring perturbation connectivity within the global network
Abundance or behavioral changes of a metabolic network component can lead to downstream genotypic and phenotypic modulations in a biological system. One important measure of robustness of a network is "connectivity" [@doi:10.1093/gigascience/giz137; @doi:10.1038/nrg.2016.87]. In a biological context, an example of connectivity, or the loss of network connectivity, is easily grasped when considering a transport reaction from the cytosol into the mitochondria of a critical metabolite [@doi:10.1186/1756-0381-4-10; @doi:10.1093/gigascience/giz137; @doi:10.1007/s41109-019-0129-0; @doi:10.1371/journal.pone.0087075]. When the transport hub is abolished, and network connectivity is lost, the critical metabolite cannot participate in the required downstream reactions. For example, our laboratory previously established the consequences of the ablation of the mitochondrial pyruvate carrier on downstream citric acid cycle processes and the corresponding increase in lactate production [@doi:10.1126/science.1218099] and serious cellular remodeling and dysfunction [@doi:10.1038/ncb3593; @doi:10.1016/j.cmet.2019.11.002].

The importance of connectivity could also be considered in a clinical context. For example, a druggable and critical metabolite may be perturbed in a particular disease context. However, if a metabolite participating in a neighboring, downstream reaction is also perturbed in a way not related to the perturbation of the first metabolite, the efficacy of the drug treatment could be severely impaired [@doi:10.1158/0008-5472.CAN-14-3256; @doi:10.1126/science.1132939; @doi:10.1186/s12918-018-0674-7; @doi:10.1038/nrg.2016.87]. To aid in the exploration of the connectivity of the biological network, we developed a connectivity module where the user can display and explore all global reactions or reactions of a specific super pathway that have at least one involved component perturbed based on either an abundance or statistical level. By doing so, proximal reactions that were perturbed will be "sewn" together to reconstruct a perturbation connectivity map in the network visualization. Perturbed, but not directly connected, reactions are displayed as single reactions in the network.

### Data vignettes
In order to demonstrate the utility of Metaboverse to the community, we used Metaboverse to analyze public and new datasets. From the vignettes provided below, we show that Metaboverse not only identifies points of interest previously described or expected, but also rapidly identifies unexpected and systematic regulatory patterns in a reaction network context.

#### 1. Perturbation of mitochondrial fatty acid synthesis across time
Mitochondrial fatty acid synthesis (mtFAS), an evolutionarily conserved pathway, has long been recognized to produce lipoic acid, a critical cofactor for several metabolic enzymes. Recent work has begun to uncover additional, important roles for this pathway. For example, we now know that mtFAS coordinates fatty acid synthesis with iron-sulfur (Fe-S) cluster biogenesis and assembly of oxidative phosphorylation complexes, and more [@doi:10.1016/j.cub.2018.08.022; @doi:10.1016/j.molcel.2018.06.039; @doi:10.7554/eLife.17828]. Additionally, this pathway has received increased physiological focus with the discovery of patients with mutations in key mtFAS enzymes [@doi:10.1016/j.ajhg.2016.09.021].

The <i>Saccharomyces cerevisiae</i> Mct1 protein is an acyltransferase responsible for the transfer of a malonyl group from malonyl-CoA to the mitochondrial acyl carrier protein (ACP). By knocking out this gene, the activity of the mtFAS pathway is abolished. In order to probe the relationship between mtFAS-related protein concentration and the effects of its perturbation on downstream metabolic processes from a systematic perspective, we used an <i>mct1</i>&Delta; mutant <i>S. cerevisiae</i> strain. We previously performed steady-state comparative proteomics in this <i>mct1</i>&Delta; yeast model 12 hours after shifting to a non-fermentable carbon source [@doi:10.1016/j.molcel.2018.06.039]. In order to demonstrate the potential of Metaboverse and build upon these previous studies, we also performed RNA-sequencing of transcript levels at 0, 3, and 12 hours after a shift to a non-fermentable carbon source, as well as steady-state metabolomics at 0, 15, 30, 60, and 180 minutes after the shift in carbon source. By layering these data onto the <i>S. cerevisiae</i> metabolic network using Metaboverse, we observed interesting respiratory signatures as expected based on previous work [@doi:10.1111/mmi.12402; @doi:10.1016/j.molcel.2018.06.039]. For example, we noticed a strong pattern in the electron transfer from ubiquinol to cytochrome C via complex III of the electron transport chain (ETC) (Figure @fig:mct1_vignette A). At the proteomics level, cytochrome C isoforms, CYC1 and CYC7 are both significantly reduced in concentration compared to wild-type cells (log<sub>2</sub>(fold change): -1.57 & -0.88; p-value: < 0.01 & < 0.01; Cohen's d: -7.99 & -3.94; respectively). This reduction in cytochrome C concentration is paired with a marked reduction in the concentration of the protein components of Complex III, which is the catalyst of the ubiquinol to cytochrome C electron transfer reaction (average log<sub>2</sub>(fold change) of all measured protein components = -2.03, where 9/11 component proteins were measured; 7/11 passed statistical significance thresholds of BH-corrected p-value <= 0.05. Range of significant log<sub>2</sub>(fold change) values: -0.34 to -3.21).  

![
  **Metaboverse identifies several reaction motifs of interest in <i>mct1</i>&Delta; cells.**
  (A) Steady-state proteomics (12 hours) overlaid on the reaction, "Electron transfer from ubiquinol to cytochrome c of complex III". (B) Steady-state proteomics (12 hours) and metabolomics at 15 minutes overlaid on TCA-related reactions. Appropriate time stamps for the data type are displayed in the upper-right hand corner of each subplot. Metabolomics and proteomics values are shown as node shading, where an increasingly blue shade indicates down-regulation and an increasingly red shade indicates up-regulation. Measured log<2> Fold Changes and statistical values for each entity are displayed below the node name. A gray node indicates a reaction. A bold gray node with purple border indicates a motif was found at this reaction for the given data type time points. Non-reaction circles indicate metabolites and squares indicate proteins or protein complexes. gray edges are core relationships between reaction inputs and outputs. Green edges indicate a catalyst. Dashed blue edges point from a metabolite component to the complex it is involved in. Dashed orange edges point from a protein component to the complex it is involved in. Protein complexes with dashed borders indicate that the values displayed on that node were inferred from the constituent protein and metabolite measurements. The shading in sub-plot A demonstrates Metaboverse's ability to show compartmentalization, whereas sub-plot B has compartmentalization shading turned off.
](./content/figures/fig_mct1.png "Square image"){#fig:mct1_vignette}

The second pattern of interest that was expected was the marked reduction of TCA cycle-related enzymes. Motif analysis within the TCA cycle identified several putative regulatory hotspots between metabolites and metabolite-protein interactions throughout the steps of the TCA cycle (Figure @fig:mct1_vignette B). However, visualizing these motifs and data across the timecourse reveals interesting patterns. For the following metabolite values, we will provide those for the 15-minute time point because the statistical strength for the discussed metabolites is strongest at this time. For example, citrate levels decrease across the metabolomics timecourse (at 15 min; log<sub>2</sub>FC: -0.-1.75, adjusted-p: 0.01; Cohen's d: -1.86), which is coincident with reduced steady-state levels of Ctp1 (log<sub>2</sub>FC: -0.64, adjusted-p: 0.03; Cohen's d: -4.73), a protein that catalyzes the transfer of citrate from the mitochondrial matrix to the cytosol [@doi:10.1074/jbc.270.8.4108]. Citrate is a key metabolite and the first step in the TCA cycle. We hypothesize that due to central carbon metabolite reductions, Ctp1 may be down-regulated in response to the decrease in citrate in order to maintain citrate concentrations within the mitochondria where it is perhaps most physiologically important in an acute respiratory metabolism context.

Another point of interest is the up-regulation of Dic1 (log<sub>2</sub>FC: 2.15, adjusted-p: < 0.01; Cohen's d: 28.97), which catalyzes the exchange of dicarboxylates like malate (at 15 min; log<sub>2</sub>FC: 2.61, adjusted-p: < 0.01; Cohen's d: 5.12) between the mitochondrial matrix and cytosol [@doi:10.1046/j.1365-2958.1999.01197.x]. Interestingly, Dic1 is essential for growth in non-fermentable carbon source media. When yeast, especially those with deficits in flux into the TCA cycle due to loss of Mct1, are switched to a non-fermentable carbon, as was done in this experiment, they might adapt by up-regulating Dic1, as we see in this situation.

We also see that although the enzymes in the TCA cycle are all reduced in concentration, several related metabolites are up-regulated across multiple time points in the dataset. It is possible, for example, that the increased fumarate levels (at 15 min; log<sub>2</sub>FC: 1.00, adjusted-p: < 0.01; Cohen's d: 3.89) are related to the reduction in active fumarate hydratase (FH tetramer, FUM1; log<sub>2</sub>FC: -1.39, adjusted-p: 0.03; Cohen's d: -4.91), as has been demonstrated previously and shown to associate with hereditary leiomyomatosis and renal cell cancer in humans [@doi:10.1038/nature19353]. These patterns would be an interesting point of further study to understand the adaptations these cells make to disruptions to the TCA cycle and their consequences.

By analyzing this multi-omics dataset using reaction motif analysis and other interactive visualization, interesting questions arise. We see several reaction motifs that are expected based on prior knowledge of this biological model, as well as other interesting behaviors worthy of further follow-up. These identified patterns demonstrate the potential Metaboverse has to act as a valuable hypothesis-generation tool, particularly with multi-omics and timecourse datasets, as well as a convenient platform for visualization and analysis of a user's dataset in the context of the metabolic or other reaction networks.

#### 2. Metabolic signatures in human lung adenocarcinomas compared to normal lung tissue
We next turned to published human lung adenocarcinoma steady-state metabolomics data [@doi:10.1158/1940-6207.CAPR-14-0329] to assess the utility of Metaboverse when analyzing this data. Lung cancer remains a leading cause of death worldwide, and a more complete understanding of the metabolism of these tumors is essential in understanding how to better treat the condition.

Consistent with the original study [@doi:10.1158/1940-6207.CAPR-14-0329], and in our recent re-analysis of the data [@doi:10.1093/gigascience/giz137], nucleotide metabolism was broadly up-regulated in adenocarcinoma samples in the motif analysis and perturbation connectivity analysis (Figure @fig:lung_vignette A). We also notice similar perturbations in xanthine (log<sub>2</sub> Fold Change: 1.35; p-value: 0.0003; Cohen's d: 0.86) (Figure @fig:lung_vignette B) and others, which are highlighted to the user quickly in the motif and perturbation connectivity analyses. However, as we previously emphasized [@doi:10.1093/gigascience/giz137], by approaching these perturbations in a reaction-centric approach, we can identify regulatory behavior that further contextualizes the data. For example, we previously highlighted through a more manual approach the up-regulation of glyceric acid (log<sub>2</sub> Fold Change: 0.50; p-value: 0.002; Cohen's d: 0.73) coupled with the proximal down-regulation of 3-Phosphoglyceric acid (log<sub>2</sub> Fold Change: -1.09; p-value: 0.004; Cohen's d: -0.68) which could indicate the activity of glycerate kinase [@doi:10.1093/gigascience/giz137]. This connection was missed in the original study, but was highlighted by the motif analysis available within Metaboverse. Activity of this sub-pathway is consequential as it has connections to serine metabolism that may be essential to generate an ideal environment for tumorigenesis (Figure @fig:lung_vignette C). Interestingly, within the measured and significant metabolites involved in the TCA cycle, we see a moderate up-regulation of malate (log<sub>2</sub> Fold Change: 0.39, adjusted-p: 0.02; Cohen's d: 0.56) and down-regulation of citrate (log<sub>2</sub> Fold Change: -0.64, adjusted-p: 0.01; Cohen's d: -0.63). This pattern is consistent with the hypothesis that the TCA cycle is being starved of carbon from pyruvate and is compensating for this deficiency by shunting carbon into the TCA cycle via glutaminolysis. This compensatory pathway, a feature in many cancers, consists of glutamine being transported into mitochondria where it is converted to alpha-ketoglutarate and enters the second half of the TCA cycle to generate reducing equivalents, ATP, and biosynthetic intermediates [@pmid:429309; @doi:10.1146/annurev-bioeng-071516-044546]. The pyruvate that would have normally been processed within the TCA cycle is probably being shunted to lactate production (no measurements were available in the dataset) or other biosynthetic products, and thus the lower citrate levels, while the distal TCA cycle is fed through glutamate (log<sub>2</sub> Fold Change: 0.47, adjusted-p: < 0.01; Cohen's d: 0.98), providing the increased malate concentration (Figure @fig:lung_vignette D) [@doi:10.1038/nrc.2016.71].

Metaboverse simplifies the analysis process and allows the user more flexibility via its interactive platform to identify both canonical and novel regulatory patterns within their data with ease. We were able to previously identify the regulatory pattern identified in Figure @fig:lung_vignette C using more manual approaches, but Metaboverse offers an automated platform for the discovery of such events, which were not identified in the original study. We anticipate that the use of Metaboverse will enable the user to extract new and exciting hypotheses that can drive their fields forward.

![
  **Metaboverse identifies nucleotide metabolism signatures in lung adenomcarcinoma metabolomics data.**
  (A) Up-regulation of several nucleotide metabolites. (B) Identification of xanthine regulation by both motif and perturbation connectivity analysis. (C) Regulatory activity between glyceric acid and 3-Phosphoglyceric acid identified by motif analysis. (D) Disruptions of TCA metabolism support canonical disruptions during adenocarcinoma development. Metabolomics values are shown as node shading, where an increasingly blue shade indicates down-regulation and an increasingly red shade indicates up-regulation. Measured log<sub>2</sub> Fold Changes and statistical values for each entity are displayed below the node name. A gray node indicates a reaction. A bold gray node with purple border indicates a motif was found at this reaction for the given data type time points. Non-reaction circles indicate metabolites, squares indicate proteins or protein complexes, and ellipses indicate gene components. gray edges are core relationships between reaction inputs and outputs. Green edges indicate a catalyst, and red edges indicate inhibitors. Dashed blue edges point from a metabolite component to the complex it is involved in. Dashed orange edges point from a protein component to the complex it is involved in. Dashed purple edges point from a gene component to its protein product. Protein complexes with dashed borders indicate that the values displayed on that node were inferred from the constituent protein and metabolite measurements.
](./content/figures/fig_lung.png "Square image"){#fig:lung_vignette}
