## Results

### Metaboverse is a dynamic, user-friendly tool for the exploration of high-throughput biological data in organism-specific pathways.

#### Overview.
We designed Metaboverse as a light-weight, self-contained app for the dynamic exploration of high-throughput biological data. The pathway curations are derived from Reactome, and as of writing, is capable of analyzing data for over ninety species. A user begins by providing a previously curated Metaboverse file, or the desired output location for a new curation and organism of interest. Next, the user provides the relevant transcriptomic, proteomic, and/or metabolomic datasets to layer onto the global reaction network of their organism of interest. These data categories can be extended to any dataset that uses the relevant mapping IDs; for example, one could provide ribosome profiling translation efficiency values mapped to the appropriate gene IDs for layering onto the network and downstream analysis. During this step, the user will also specify a few experimental parameters for consideration during downstream analysis and visualization. Following these user inputs, the organism's network, data processing, and motif analysis (discussed further below) is curated and a curation file is output for future analysis (see Figure @fig:package_overview).

![
  **Metaboverse package overview.**
  Illustration comparing the data input, processing, and exploration steps of the Metaboverse package.
](./content/figures/package_overview.png "Square image"){#fig:package_overview}

#### Handling data sparsity within the biological global network.
Missing data points, particularly in metabolomics experiments, are frequent and can make analysis of pathways and identification of regulatory patterns in the network challenging.  While thousands of metabolites are known to participate in human metabolism, the current state of the technology for determining which mass spectra belong to which metabolite can be challenging and often results in a limited number of data points being quantified. These can lead to gaps in the metabolic network which can be challenging to explore and analyze. We therefore developed a reaction compression algorithm (detailed more in the Methods section) that collapses up to three reactions with missing data points if they can be bridged with known data on the distal ends of the reaction path. These reactions, or pseudo-reactions are visually distinct to show users where reactions were collapsed and understand what that pseudo-reaction summarizes.

#### Rapid identification of interesting regulatory patterns in the reaction network.
Following network curation, the user can visualize available reaction motifs identified across the global reaction network. In a computational science context, a motif is a re-occurring pattern in network structure, or the organization of network entities and their relationships to one another. However, with -omic data, we are more interested in identifying patterns in expression or abundance of genes, proteins, and metabolites. Previous work by Checkik, et. al. introduced the concept of "activity motifs", where instead of identifying motifs based on network structure, they were identified using the expression characteristics of nodes in transcription factor binding signaling networks [@doi:10.1038/nbt.1499]. We adapted this methodology to identify and interactively display interesting regulatory hotspots within the global reaction network. For example, a reaction's inputs may exhibit increased levels and its outputs low levels, indicating a regulatory event occurring at the reaction.

In Metaboverse, we define a motif as a regulatory pattern identified across a reaction or pseudo-reaction. Metaboverse contains a library of default motifs to search the network for, and users can define custom motifs they would like to identify across the global network (please refer to the documentation for more details on available motifs [@url:https://metaboverse.readthedocs.io/en/latest]). Metaboverse will search the global network from a pre-defined library of regulatory patterns (regulatory hotspots, or motifs) and return a graphical stamp view of conserved patterns. These motifs by default are ordered by the statistical values associated with the entity components of the motif, but can be ordered by the strength of the motif based on the magnitude of change. Statistical values associated with each measurement for a measured entity as provided by the user are used to weight these motifs and prioritize returned results. We use a three-tiered sorting strategy when sorting by the associated statistical values. The highest prioritized motifs are those where the relevant components on either side of the reaction (inputs vs. outputs) that determined the reaction motif are statistically significant. Of these motifs from the first tier, motifs are sorted by lowest to highest cumulative p-value or other relevant statistical value. In the second sorting tier, reactions where at least one side of the reaction had a statistically significant motif component are sorted by statistical strength. Finally, all other motifs are sorted by the cumulative statistical value for the motif components.

Reaction motifs are visually distinct during network visualization to quickly draw the user to these interesting patterns. For a given pattern, the user can then explore each pathway this particular motif is found in. Motif analysis of the global regulatory network will allow users to rapidly identify interesting features within the data, particularly patterns between canonical pathways or in other pathways that may seem tangential in their research, but are instead rather striking based on these contextualized analytical method. In the data vignettes below, we demonstrate this utility further. Users will also be able to design their own patterns through an interactive pattern drawing tool, and even design specific scenarios that are cognizant of feature type (in progress). For example, one might be interested in a pattern where a protein displays higher expression, but the resulting metabolite is decreased. This could then be coded interactively into the Metaboverse framework by the user.

#### Dynamic visualization of organism-specific reaction pathways.
Following curation of the global network as described above, the user can manually search individual canonical pathways or individual entities and their reaction neighborhoods. For a given selection, all relevant reactions that are annotated as a part of that pathway will be graphed, along with their core input (reactant) and output (product) components. In addition to these core elements, known catalysts and inhibitors are included, as well as the component proteins, genes, and metabolites that form functional complexes involved in a particular reaction. Labels can be toggled on or off in the display, and the user can switch between viewing the values or statistics associated with each data point with their relevant color mapping. In cases where a gene value is known, but its protein value is unmeasured, the protein value will be inferred using aggregated gene component values. The same is then done for functional complexes using their inferred or known component values. Relevant pathway and analytical metadata is also displayed. Other information, such as identified motifs found in the pathway can also be found and expanded for further exploration in a new window. Aids for visualization are also available, such as the ability to remove nodes from visualization that contain a high number of relationships to other network features such as that these nodes, which act as hubs in the network, do not lead to cluttered representations of the network. Often, these hub nodes are ubiquitous features such as water and proton which may be of limited interest to the user during data visualization. Compartment domains are also graphed to include a relevant reactions and their components that occur in a given cellular compartment.

#### Visualization of downstream effects of network perturbations.
Users may be interested a particular metabolite or protein and the downstream effects its perturbation has on related pathways. By double-clicking a node of interest, or by selecting the entity name from the drop-down menu, the user can explore all downstream effects across all pathways in the global network. The user can also define how many neighborhoods to display such that one can visualize two or more reaction steps downstream of the selected entity. This functionality moves the analysis away from our traditional, strictly defined pathway approach to analysis, and helps contextualize the far-reaching effects changes in metabolism or other biological systems can have across classical pathways.

#### Exploring perturbation connectivity within the global network
Abundance or behavioral changes of a metabolic network component can lead to downstream genotypic and phenotypic modulations in a biological system. One important measure of robustness of a network is "connectivity". In a biological context, an example of connectivity, or the loss of network connectivity, is easily grasped when considering a transport reaction from the cytosol into the mitochondria of a critical metabolite [@doi:10.1186/1756-0381-4-10; @doi:10.1093/gigascience/giz137; @doi:10.1007/s41109-019-0129-0; @doi:10.1371/journal.pone.0087075]. When the transport hub is abolished and network connectivity is lost, the critical metabolite cannot participate in the required downstream reactions. For example, an example from our laboratory established the consequences of the ablation of the mitochondrial pyruvate carrier on downstream citric acid cycle processes and the corresponding increase in lactate production [@doi:10.1126/science.1218099] and leads to serious cellular complications and dysfunction [@doi:10.1038/ncb3593; @doi:10.1016/j.cmet.2019.11.002].

The importance of connectivity could also be considered in a treatment context. For example, a druggable and critical metabolite may be perturbed in a particular disease context. However, if a metabolite participating in a neighboring, downstream reaction is also perturbed in a way not related to the perturbation of the first metabolite, the efficacy of the drug treatment could be severely impaired [@doi:10.1158/0008-5472.CAN-14-3256; @doi:10.1126/science.1132939; @doi:10.1186/s12918-018-0674-7]. To aid in exploration of the connectivity of the biological network, we also developed a connectivity module where users can display all reactions that have at least one involved component perturbed based on either an abundance or statistical level. By doing so, when the graph is constructed, proximal reactions that were perturbed independently will be "sewn" together to reconstruct a perturbation connectivity map in the user's model.

### Data vignettes
In order to demonstrate the added utility of Metaboverse to the community that is not currently available in other tools, we used Metaboverse to analyze a series of public and new datasets. From the vignettes provided below, we show that Metaboverse not only is able to identify points of interest previously described or expected, but can rapidly identify for the user unexpected and systematic regulatory patterns in a reaction network context.

#### 1. Perturbation of mitochondrial fatty acid synthesis across time reveals expected and unexpected reaction motifs.
Mitochondrial fatty acid synthesis (mtFAS) acts as an important metabolic pathway long known to function to produce lipoic acid, a critical cofactor for several metabolic enzymes. Recent work has begun to uncover additional, important roles for this pathway. For example, we now know that mtFAS can perform metabolic nutrient sensing roles to coordinate lipoic acid synthesis with iron sulfur (FeS) cluster biogenesis, assembly of oxidative phosphorylation complexes, and more [@doi:10.1016/j.cub.2018.08.022; @doi:10.7554/eLife.17828]. Additionally, this pathway has received increased physiological focus with the discovery of patients with mutations in key mtFAS enzymes [@doi:10.1016/j.ajhg.2016.09.021].

<i>MCAT</i>, or <i>MCT1</i>in <i>Saccharomyces cerevisiae</i>, is an acyltransferase responsible for the transfer of a malonyl group from malonyl-CoA to <i>NDUFAB1</i>, the mitochondrial acyl carrier protein. By knocking out this gene, one can simulate the effects of a disruption in mtFAS.

In order to probe the relationship between mtFAS-related protein concentration and the effects of its perturbation on downstream metabolic processes from a systematic perspective, we used a <i>mct1&Delta;</i> model in <i>Saccharomyces cerevisiae</i> to measure steady state proteomics and transcriptomics at 12 hours after a shift to a non-fermentable carbon source, as well as steady state metabolomics at 0, 15, 30, 60, and 180 minutes after this shift in carbon source. By layering these data onto the <i>Saccharomyces cerevisiae</i> metabolic network using Metaboverse, we observed interesting respiratory signatures as expected based on our previous work [@doi:10.1016/j.cub.2018.08.022; @doi:10.7554/eLife.17828]. For example, using Metaboverse, we noticed a strong pattern in the electron transfer from ubiquinol to cytochrome C of complex III of the electron transport chain (ETC) (Figure @fig:mct1_vignette A). At the proteomics level, cytochrome C component proteins, CYC1 and CYC7 are both significantly reduced in concentration compared to wild-type cells (log<sub>2</sub>(fold change): -1.57 & -0.88; p-value: < 0.01 & < 0.01; respectively). This reduction in cytochrome C concentration is paired with a marked reduction in concentration of the protein components of cytochrome C reductase, which is a catalyst of this electron transfer reaction (average log<sub>2</sub>(fold change) of all measured protein components = -2.03, where 9/11 component proteins were measured. Range of log<sub>2</sub>(fold change) values: -0.34 to -3.21; all p-values below 0.01).   

![
  **Metaboverse identifies several reaction motifs of interest in <i>mct1&Delta;</i> cells.**
  (A) (B) (C) . All panels shown used the 180 min. metabolomics values for node shading.
](./content/figures/fig_mct1.png "Square image"){#fig:mct1_vignette}

A second pattern of interest that we expect in this model is the conversion of Succinate to Fumarate by the Succinate dehydrogenase complex (SDH, or Complex II of the ETC) (Figure @fig:mct1_vignette B). We know from previous work that complex assembly and mtFAS are intricately connected [@doi:10.1016/j.cub.2018.08.022; @doi:10.1101/2020.05.09.086199]. However, this particular behavior when pairing the temporal metabolomics data with the steady-state proteomics data is interesting and acts as a potential jumping-off point for further follow up experimentation. Across each time point, we see a steady accumulation of fumarate (log<sub>2</sub>(fold change) >= 1). We pair this with the proteomics data and observe a strong reduction in both SDH and FUM1, which encodes the fumarase needed to convert fumurate to malate in the TCA cycle (-0.49 to -2.96). It is possible that the reduction of FUM1 and build-up of fumurate is acting as a bottleneck and signaling a reduction in Complex II component production and assembly. However, as measurements for succinate and malate are missing, it is difficult to clarify this regulation more without further follow-up experiments. An additional hypothesis may involve SFC1, the mitochondrial succinate-fumarate transporter, which we previously noticed from the analysis of the proteomics data was reduced in concentration 2.5-fold. Unfornately, at this time this transport reaction is not annotated in the Reactome database for <i>Saccharomyces cerevisiae</i>, which raises an important goal for the Metaboverse project, which is to augment these pathway representations to fill in the gaps of knowledge using additional sources. However, with the prior knowledge, it is interested to notice this patten with fumarate as the build-up of fumarate may reflect the inability of the mitochondria to export fumarate.

A final observation of interest from analysis through Metaboverse is that of citrate and isocitate. Normally, the TCA cycle generates citrate from malate to oxaloacetate via MDH1, then oxaloacetate to citrate via CIT1 and CIT3. Building off the observations discussed above, we expect lower levels of malate as due to decreased MDH1. We also observe lower levels of CIT1 (-1.46) and reciprocal decreases in citrate that worsens across the time-course. However, at 30 minutes post-carbon source shift, isocitrate levels are decreased by -1.38 log<sub>2</sub>(fold change) (p-value < 0.01). By 60 and 180 minutes, there is no significant change in isocitrate levels compared to wild-type cells, while citrate levels remain significantly reduced (Figure @fig:mct1_vignette C). Does this indicate some sort of adaptation to a respiratory-required environment with ETC-impaired cells?

By analyzing this multi-omics dataset using reaction motif analysis and other interactive visualization, interesting questions arise. We see several reaction motifs that are expected based on prior knowledge of this biological model, as well as other interesting behaviors worthy of further follow-up. This demonstrates the potential Metaboverse has to act as a valuable hypothesis-generation tool, as well as a convenient platform for visualization and analysis of a user's dataset in the context of the metabolic or other reaction networks.
